<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>genome_anno | TianzhenWu' Blog</title><meta name="author" content="吴 田 振"><meta name="copyright" content="吴 田 振"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="主题：基因组注释相关参考： 基因组注释（一）：重复序列注释 | 生信技工 (yanzhongsino.github.io) 使用RepeatModeler从头预测基因组重复序列 | BioChen 博客 【基因组注释】RepeatMasker和RepeatModeler安装、配置与运行避坑 - 简书 (jianshu.com) 基因组重复序列注释repeatmask&amp;repeatmodel">
<meta property="og:type" content="article">
<meta property="og:title" content="genome_anno">
<meta property="og:url" content="https://wu-tz.github.io/2024/07/17/genome-anno/index.html">
<meta property="og:site_name" content="TianzhenWu&#39; Blog">
<meta property="og:description" content="主题：基因组注释相关参考： 基因组注释（一）：重复序列注释 | 生信技工 (yanzhongsino.github.io) 使用RepeatModeler从头预测基因组重复序列 | BioChen 博客 【基因组注释】RepeatMasker和RepeatModeler安装、配置与运行避坑 - 简书 (jianshu.com) 基因组重复序列注释repeatmask&amp;repeatmodel">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://wu-tz.github.io/img/orca-295051_1280.png">
<meta property="article:published_time" content="2024-07-17T13:06:54.000Z">
<meta property="article:modified_time" content="2024-07-17T13:11:08.259Z">
<meta property="article:author" content="吴 田 振">
<meta property="article:tag" content="genome注释">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wu-tz.github.io/img/orca-295051_1280.png"><link rel="shortcut icon" href="/img/w-833386_640.png"><link rel="canonical" href="https://wu-tz.github.io/2024/07/17/genome-anno/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":false,"top_n_per_article":1,"unescape":false,"languages":{"hits_empty":"找不到您查询的内容：${query}","hits_stats":"共找到 ${hits} 篇文章"}},
  translate: {"defaultEncoding":2,"translateDelay":0,"msgToTraditionalChinese":"繁","msgToSimplifiedChinese":"简"},
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '天',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'genome_anno',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2024-07-17 21:11:08'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
    win.getCSS = (url,id = false) => new Promise((resolve, reject) => {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = url
      if (id) link.id = id
      link.onerror = reject
      link.onload = link.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        link.onload = link.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(link)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 5.4.2"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/orca-295051_1280.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">29</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">18</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 归档</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/messageboard/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/img/rainy-days-4481077_1280.jpg')"><nav id="nav"><span id="blog-info"><a href="/" title="TianzhenWu' Blog"><img class="site-icon" src="/img/w-833386_640.png"/><span class="site-name">TianzhenWu' Blog</span></a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search" href="javascript:void(0);"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 归档</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/messageboard/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">genome_anno</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2024-07-17T13:06:54.000Z" title="发表于 2024-07-17 21:06:54">2024-07-17</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2024-07-17T13:11:08.259Z" title="更新于 2024-07-17 21:11:08">2024-07-17</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">7.2k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>30分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="genome_anno"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h3 id="主题：基因组注释"><a href="#主题：基因组注释" class="headerlink" title="主题：基因组注释"></a>主题：基因组注释</h3><p>相关参考：</p>
<p><a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/08/02/omics_genome.annotation_repeat/">基因组注释（一）：重复序列注释 | 生信技工 (yanzhongsino.github.io)</a></p>
<p><a target="_blank" rel="noopener" href="https://www.biochen.org/cn/blog/2021/%E4%BD%BF%E7%94%A8RepeatModeler%E4%BB%8E%E5%A4%B4%E9%A2%84%E6%B5%8B%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%87%8D%E5%A4%8D%E5%BA%8F%E5%88%97/">使用RepeatModeler从头预测基因组重复序列 | BioChen 博客</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c83ad97d1c7c">【基因组注释】RepeatMasker和RepeatModeler安装、配置与运行避坑 - 简书 (jianshu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/147585710">基因组重复序列注释repeatmask&amp;repeatmodeler - 知乎 (zhihu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/a0dee7c5fdef">非模式生物重复序列注释 RepeatModeler2+RepeatMasker4 - 简书 (jianshu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://phantom-aria.github.io/2023/03/16/a.html">基因组注释（2）——散在重复序列注释 - 我的小破站 (phantom-aria.github.io)</a></p>
<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/379464361">使用AUGSTUS+Geneid+GeneMark+GeMoMa+GenomeThreader+Exonerate进行基因结构预测 - 知乎 (zhihu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/b75b8e253552">基因组结构注释软件列表 - 简书 (jianshu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://phantom-aria.github.io/2023/03/12/a.html">基因组注释（1）——串联重复序列注释 - 我的小破站 (phantom-aria.github.io)</a></p>
<h4 id="1-重复序列注释"><a href="#1-重复序列注释" class="headerlink" title="1 重复序列注释"></a>1 重复序列注释</h4><h5 id="1-1-重复序列的主要分类"><a href="#1-1-重复序列的主要分类" class="headerlink" title="1.1  重复序列的主要分类"></a>1.1  重复序列的主要分类</h5><p>重复序列包括串联重复(tandem repeats)和散布重复(dispersed repeats)。前者按照重复的长度可划分为卫星序列(&gt;100bp)、小卫星序列(&gt;10bp and&lt;100bp)和微卫星序列(SSR，&lt;10bp)，没有固定的长度划分标准。后者也叫做转座子(transposable element，TE)，包括DNA转座子(DNA transposons，也叫做Ⅱ型转座子)和RNA转座子(Retrotransposon，也叫做Ⅰ型转座子、反转录转座子)，其中RNA转座子包括长末端重复序列LTR、长散布核元件LINE和短散布核元件SINE。</p>
<h5 id="1-2-注释软件"><a href="#1-2-注释软件" class="headerlink" title="1.2 注释软件"></a>1.2 注释软件</h5><p>TRF：检测序列中串联重复序列，全称为Tandem Repeat Finder；</p>
<p>GMATA：分析基因组中的SSR序列；</p>
<p>RepeatScout：从头注释重复序列的核心组件；</p>
<p>RECON：从头注释重复序列的核心组件；</p>
<p>RepeatModeler：主要针对非模式物种的从头注释；RepeatMasker version 4.1.5</p>
<p>RepeatMasker：通过重复序列数据库（包括Dfam和Repbase等）和从头注释得到的数据库进行注释得到整合的结果；</p>
<p><em>文献中软件使用的案例：</em></p>
<p>①The rough-toothed dolphin genome provides new insights into the genetic mechanism of its rough teeth (Huang, et al., 2023)</p>
<img src="/2024/07/17/genome-anno/image-20240325141415995.png" class title="image-20240325141415995">

<p>②Chromosome-level genome assembly of hadal snailfish reveals mechanisms of deep-sea adaptation in vertebrates (Xu, et al., 2023)</p>
<img src="/2024/07/17/genome-anno/image-20240325141612852.png" class title="image-20240325141612852">

<p>③Genome assembly of the deep-sea coral Lophelia pertusa  (Herrera, et al., 2022)</p>
<img src="/2024/07/17/genome-anno/image-20240325142112181.png" class title="image-20240325142112181">

<p>④The FirstDraft Genome of a Cold-Water Coral Trachythela sp. (Alcyonacea: Stolonifera: Clavulariidae) (Zhou, et al., 2020)</p>
<img src="/2024/07/17/genome-anno/image-20240325142530238.png" class title="image-20240325142530238">

<h5 id="1-3-用于同源注释的数据库"><a href="#1-3-用于同源注释的数据库" class="headerlink" title="1.3 用于同源注释的数据库"></a>1.3 用于同源注释的数据库</h5><p>①Dfam：RepeatMasker自带的库，提供了大量已知的重复元件家族的信息；</p>
<p>②Repbase：用于存储和管理已知重复序列的数据库，在网上可以下载到2018年的版本，更新的版本则需要注册费下载；</p>
<p>下载并配置repbase库，进入到&#x2F;home&#x2F;tianzhen&#x2F;miniconda3&#x2F;envs&#x2F;buscopy3.9&#x2F;share&#x2F;RepeatMasker&#x2F;目录下，运行配置命令</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl ./configure</span><br></pre></td></tr></table></figure>

<img src="/2024/07/17/genome-anno/image-20240325102127133.png" class title="image-20240325102127133">

<p>③dc20170127-rb20170127数据库：不太了解，仅在几篇推文中看到过；</p>
<p>④通过RepeatModeler对自己的物种进行从头注释的数据库；</p>
<h5 id="1-4-软件安装与使用"><a href="#1-4-软件安装与使用" class="headerlink" title="1.4 软件安装与使用"></a>1.4 软件安装与使用</h5><p>①首先注释串联重复序列，利用GMATA和TRF两种软件。</p>
<p>GMATA安装非常简单，没有conda途径。下载链接<a target="_blank" rel="noopener" href="https://sourceforge.net/projects/gmata/%EF%BC%8C%E8%A7%A3%E5%8E%8B%E5%90%8E%E4%B8%BA%E4%B8%80%E7%B3%BB%E5%88%97perl%E8%84%9A%E6%9C%AC%E5%92%8C%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6%E7%AD%89%EF%BC%8C%E9%A6%96%E5%85%88%E4%BF%AE%E6%94%B9default_cfg.txt%E6%96%87%E4%BB%B6%E4%B8%AD%E5%8F%82%E6%95%B0%E8%AE%BE%E7%BD%AE%EF%BC%8C%E5%B0%86[set]:doprimer_smt%E3%80%81[set]:elctPCR%E5%92%8C[set]:mk2gff3%E7%9A%84ModulRun">https://sourceforge.net/projects/gmata/，解压后为一系列perl脚本和配置文件等，首先修改default_cfg.txt文件中参数设置，将[set]:doprimer_smt、[set]:elctPCR和[set]:mk2gff3的ModulRun</a> &#x3D; Y改为ModulRun &#x3D; N，三者是设计SSR引物和模拟PCR的时候需要调用的，若设置为Y还得需要下载其他依赖。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#在default_cfg.txt文件默认参数中，每个SSR-motif的标准是：最小长度为2，最大长度为10，最少重复5次</span></span><br><span class="line">perl gmata.pl -c default_cfg.txt -i /project/tianzhenWu/z189/03genome_annotation/purged.fa</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用trf（Tandem Repeats Finder, Version 4.09）预测串联重复序列，参数如下</span></span><br><span class="line">conda install bioconda::trf</span><br><span class="line">trf ../purged.fa 2 7 7 80 10 50 500 -f -d -h -r</span><br><span class="line"></span><br><span class="line"><span class="comment">#提取trf结果为gff3文件</span></span><br><span class="line">perl ~/scripts/repeat_to_gff.pl purged.fa.2.7.7.80.10.50.500.dat</span><br></pre></td></tr></table></figure>

<p>结果解读：GMATA和TRF的结果具有重叠性，前者的默认参数预测了以2-10bp为单位的串联重复（SSR），后者的默认参数则会预测2000bp以下为单位的串联重复，这里我后续的分析并非以SSR为主，因此只用了TRF的结果。TRF结果的统计见推文<a target="_blank" rel="noopener" href="https://phantom-aria.github.io/2023/03/12/a.html">基因组注释（1）——串联重复序列注释 - 我的小破站 (phantom-aria.github.io)</a>，内含脚本，写得非常详细 [点赞]。</p>
<p>②针对散在重复序列，结合从头注释和同源注释的思路。先利用RepeatModeler进行自我比对，建立本地数据库，然后利用RepeatMasker根据公共数据库近缘物种的重复序列和本地自建的数据库进行重复序列的从头注释。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#利用conda安装RepeatModeler的2.0.5版本，根据其依赖包的要求配置python环境，本文以python3.9版本为例</span></span><br><span class="line">conda install repeatmodeler=2.0.5</span><br><span class="line"></span><br><span class="line"><span class="comment">#从头预测转座子</span></span><br><span class="line">BuildDatabase -name z189 -engine ncbi ../purged.fa</span><br><span class="line">RepeatModeler -pa 20 -database z189 -engine ncbi</span><br></pre></td></tr></table></figure>

<p>RepeatModeler运行五轮结束（可能也会运行6轮），日志文件如下：</p>
<img src="/2024/07/17/genome-anno/image-20240327094818877.png" class title="image-20240327094818877">

<p>结果主要包括z189-families.fa、z189-families.stk和RM_29354.TueMar261429412024目录，目录下包括各种过程文件。</p>
<p>利用RepeatModeler的结果z189-families.fa文件运行RepeatMasker，此时运用-lib参数指定目标文件，然而当利用重复序列数据库运行RepeatMasker时，则运用-species参数指定目标物种，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">RepeatMasker ../purged.fa -lib z189-families.fa -e ncbi -pa 40 -poly -html -gff -dir ./denove/</span><br></pre></td></tr></table></figure>

<p>利用重复序列数据库近缘物种的重复序列运行RepeatMasker，先找到目标物种的近缘种是否在数据库中，然后提取近缘物种的重复序列，形成近缘物种重复序列数据库，将其与上面RepeatModeler产生的自身比对重复序列数据库合并，形成最终用于从头注释的库repeat_for_anno.fasta。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">python famdb.py -i Libraries/RepeatMaskerLib.h5 lineage -ad Anthozoa  <span class="comment">#查找信息</span></span><br><span class="line"></span><br><span class="line">python famdb.py -i Libraries/RepeatMaskerLib.h5 families -f embl -a -d Anthozoa &gt; Anthozoa.embl <span class="comment">#将信息存储为embl格式文件</span></span><br><span class="line"></span><br><span class="line">buildRMLibFromEMBL.pl Anthozoa.embl &gt; Anthozoa.fasta <span class="comment">#创建近缘物种重复序列</span></span><br><span class="line"></span><br><span class="line">cat /home/tianzhen/miniconda3/envs/buscopy3.9/share/RepeatMasker/Anthozoa.fasta ../z189-families.fa &gt; repeat_for_anno.fasta <span class="comment">#将RepeatMasker重复序列库和近缘物种重复序列库合并</span></span><br><span class="line"></span><br><span class="line"><span class="comment">##进行软屏蔽，并输出重复序列表，加上-a参数输出比对文件</span></span><br><span class="line">RepeatMasker -a -xsmall -nolow -norna -html -gff -dir ./xsmall_output -lib repeat_for_anno.fasta -e ncbi -pa 50 -poly ../../purged.fa</span><br></pre></td></tr></table></figure>



<p>the neutral mutation rate (μ) was calculated using r8s v1.8.1</p>
<h4 id="2-非编码RNA注释"><a href="#2-非编码RNA注释" class="headerlink" title="2 非编码RNA注释"></a>2 非编码RNA注释</h4><h5 id="2-1-非编码rRNA分类"><a href="#2-1-非编码rRNA分类" class="headerlink" title="2.1 非编码rRNA分类"></a>2.1 非编码rRNA分类</h5><p>非编码RNA (ncRNA)包括rRNA, tRNA和其他小RNA (miRNA、snRNA等)。更详细的分类可以在Rfam网站进行查询<a target="_blank" rel="noopener" href="https://rfam.org/search#tabview=tab5">Rfam: Search Rfam</a>。</p>
<h5 id="2-2-注释软件"><a href="#2-2-注释软件" class="headerlink" title="2.2 注释软件"></a>2.2 注释软件</h5><p>①Rnammer：一种用于预测原核生物和真核生物基因组中的RNA基因的工具，能够识别并定位16S&#x2F;18S、5S和23S&#x2F;28S rRNA基因序列，该软件利用隐马尔可夫模型（HMM）来识别RNA基因的保守结构特征，从而进行预测和定位。</p>
<p>②tRNAscan-SE：用于预测原核生物和真核生物基因组中转运RNA（tRNA）基因的工具，其结合了序列比对和结构预测的方法，能够准确地识别tRNA基因序列并预测其二级结构。tRNAscan-SE软件具有较高的准确性和灵敏度，为基因组注释、生物信息学研究和进化分析提供重要的信息。</p>
<p>③INFERNAL：运用Cmsan程序查找各类非编码RNA，包括miRNA、snRNA等。</p>
<h5 id="2-3-用于注释的数据库"><a href="#2-3-用于注释的数据库" class="headerlink" title="2.3 用于注释的数据库"></a>2.3 用于注释的数据库</h5><p>RFAM数据库是一个专门用于存储和注释非编码RNA序列的数据库<a target="_blank" rel="noopener" href="https://rfam.org/">Rfam: The RNA families database</a>。可以在这个页面<a target="_blank" rel="noopener" href="https://rfam.org/search#tabview=tab5">Rfam: Search Rfam</a>查看不同非编码RNA的分类，以及通过sequence search功能来查看我们所感兴趣的序列片段是否属于某一个非编码RNA家族。</p>
<h5 id="2-4-软件安装与使用"><a href="#2-4-软件安装与使用" class="headerlink" title="2.4 软件安装与使用"></a>2.4 软件安装与使用</h5><h6 id="2-4-1-利用Infernal软件注释ncRNA"><a href="#2-4-1-利用Infernal软件注释ncRNA" class="headerlink" title="2.4.1 利用Infernal软件注释ncRNA"></a>2.4.1 利用Infernal软件注释ncRNA</h6><p>参考<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/04/22/omics_genome.annotation_ncRNA/">基因组注释（四）：非编码RNA的注释-用Infernal软件对Rfam 12进行RNA注释 | 生信技工 (yanzhongsino.github.io)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install infernal=1.1.5</span><br></pre></td></tr></table></figure>

<p>在Rfam网站<a target="_blank" rel="noopener" href="https://rfam.org/">Rfam: The RNA families database</a>下载RNA family数据库h：版本为Rfam 14.10 (November 2023, 4170 families)，下载配套的clanin文件</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">wget -c https://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/Rfam.cm.gz</span><br><span class="line"></span><br><span class="line">wget -c https://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/Rfam.clanin</span><br><span class="line"></span><br><span class="line">gunzip Rfam.cm.gz <span class="comment">#解压</span></span><br><span class="line"></span><br><span class="line">cmpress Rfam.cm  <span class="comment">#建立索引并生成Rfam.cm.i1f, Rfam.cm.i1i, Rfam.cm.i1m, Rfam.cm.i1p</span></span><br><span class="line"></span><br><span class="line">cmscan -Z 1650 --cut_ga --rfam --nohmmonly --fmt 2 --tblout sample.tblout -o sample.result --clanin Rfam.clanin Rfam.cm ../purged.fa.masked <span class="comment">#-Z参数根据基因组大小来定，基因组大小的2倍，以Mb单位选一个整数</span></span><br><span class="line"></span><br><span class="line">perl infernal-tblout2gff.pl --cmscan --fmt2 sample.tblout &gt;sample.infernal.ncRNA.gff3 <span class="comment">#将结果转成gff3文件</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#统计各类ncRNA总数</span></span><br><span class="line">awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;;&#125;&#123;if(FNR==1) print &quot;target_name\taccession\tquery_name\tquery_start\tquery_end\tstrand\tscore\tEvalue&quot;; if(FNR&gt;2 &amp;&amp; $20!=&quot;=&quot; &amp;&amp; $0!~/^#/) print $2,$3,$4,$10,$11,$12,$17,$18; &#125;&#x27;</span> sample.tblout &gt;sample.tblout.xls</span><br><span class="line"></span><br><span class="line"><span class="comment">#在rfam官网下载所有的Entry types</span></span><br><span class="line">cat rfam.txt | awk <span class="string">&#x27;BEGIN &#123;FS=OFS=&quot;\t&quot;&#125;&#123;split($3,x,&quot;;&quot;);class=x[2];print $1,$2,$3,$4,class&#125;&#x27;</span> &gt; rfam_anno.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">#统计结果</span></span><br><span class="line">awk <span class="string">&#x27;BEGIN&#123;OFS=FS=&quot;\t&quot;&#125;ARGIND==1&#123;a[$2]=$5;&#125;ARGIND==2&#123;type=a[$1]; if(type==&quot;&quot;) type=&quot;Others&quot;; count[type]+=1;&#125;END&#123;for(type in count) print type, count[type];&#125;&#x27;</span> rfam_anno.txt sample.tblout.xls &gt;sample.ncRNA.statistic</span><br><span class="line"></span><br><span class="line"><span class="comment">#统计具体的分类信息，比如miRNA</span></span><br><span class="line">grep <span class="string">&quot;miRNA&quot;</span> rfam_anno.txt |cut -f1 &gt;miRNA.tem</span><br><span class="line">grep -f miRNA.tem sample.tblout.xls &gt;miRNA.txt</span><br><span class="line">awk <span class="string">&#x27;&#123;sum += (int($5) - int($4) &gt;= 0 ? int($5) - int($4) : int($4) - int($5)) + 1&#125; END &#123;print sum&#125;&#x27;</span> miRNA.txt <span class="comment">#打印miRNA序列总长数值到屏幕</span></span><br></pre></td></tr></table></figure>

<p>本部分详细解读请查阅生信技工的博客，非常详细且无报错<a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2022/04/22/omics_genome.annotation_ncRNA/">基因组注释（四）：非编码RNA的注释-用Infernal软件对Rfam 12进行RNA注释 | 生信技工 (yanzhongsino.github.io)</a></p>
<h6 id="2-4-2-利用rnammer-注释rRNA"><a href="#2-4-2-利用rnammer-注释rRNA" class="headerlink" title="2.4.2 利用rnammer 注释rRNA"></a>2.4.2 利用rnammer 注释rRNA</h6><p>rnammer软件不能通过conda进行安装，目前只允许院校研究人员通过单位邮件来申请，申请地址为<a target="_blank" rel="noopener" href="https://services.healthtech.dtu.dk/services/RNAmmer-1.2/">RNAmmer 1.2 - DTU Health Tech - Bioinformatic Services</a>，提交后立即发送下载链接到邮箱，下载链接4小时内有效，下载链接如下。</p>
<img src="/2024/07/17/genome-anno/image-20240328203933759.png" class title="image-20240328203933759">

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget -c rnammer-1.2.Unix.tar.gz <span class="comment">#下载软件</span></span><br><span class="line"></span><br><span class="line">tar zxf rnammer-1.2.Unix.tar.gz <span class="comment">#解压软件包，其中rnammer文件为一个perl脚本，执行该脚本便可以预测rRNA，但是需要修改其中的两个绝对路径，以便脚本运行时能够找到相应的程序。而修改的绝对路径包括本perl脚本本身的安装目录以及序列检索所需要的Hmmer程序。rnammer要求hmmer版本为2.3.2，若已安装最新版本的hmmer，则需要重新安装hmmer-2.3.2。另外其他不同推文表示该软件依赖perl模块，我利用conda安装了perl-xml-simple便可运行成功，也有推文表示需要perl4版本下的模块Perl4::CoreLibs，这个我没有安装，也可运行成功。</span></span><br></pre></td></tr></table></figure>

<p>安装hmmer2.3.2和perl模块</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">conda install perl-xml-simple <span class="comment">#安装perl-xml-simple</span></span><br><span class="line"></span><br><span class="line">wget http://eddylab.org/software/hmmer/hmmer-2.3.2.tar.gz <span class="comment">#下载hmmer-2.3.2</span></span><br><span class="line"></span><br><span class="line">tar -xf hmmer-2.3.2.tar.gz</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> hmmer-2.3.2</span><br><span class="line"></span><br><span class="line">./configure</span><br><span class="line"></span><br><span class="line">make</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> src;hmmsearch <span class="comment">#打印帮助文件则表示安装成功</span></span><br></pre></td></tr></table></figure>

<p>在rnammer文件的目录下修改rnammer中的绝对路径</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">perl -p -i -e <span class="string">&#x27;s/(my \$INSTALL_PATH).*/$1 = \&quot;\/project\/tianzhenWu\/software\/rnammer\&quot;;/&#x27;</span> rnammer <span class="comment">#将\/project\/tianzhenWu\/software\/rnammer换为rnammer的目录</span></span><br><span class="line"></span><br><span class="line">perl -p -i -e <span class="string">&#x27;s/^(\s+\$HMMSEARCH_BINARY).*/$1 = \&quot;\/project\/tianzhenWu\/software\/hmmer-2.3.2\/src\/hmmsearch\&quot;;/&#x27;</span> rnammer <span class="comment">#将\/project\/tianzhenWu\/software\/hmmer-2.3.2\/src\/换位hmmsearch的目录</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#另外，修改该目录下core-rnammer的多线运行限制，可以缩短运行时间</span></span><br><span class="line">perl -p -i -e <span class="string">&#x27;s/--cpu 1//g&#x27;</span> core-rnammer</span><br><span class="line"></span><br><span class="line"><span class="comment">#运行脚本，注释rRNA</span></span><br><span class="line">perl /project/tianzhenWu/software/rnammer/rnammer -S euk -multi -m lsu,ssu,tsu -gff rRNA.gff2 -f rRNA.fasta -h rRNA.hmmreport -xml rRNA.xml /project/tianzhenWu/z189/03genome_annotation/03ncrna/purged.fa.masked</span><br></pre></td></tr></table></figure>

<p>此部分主要参考<a target="_blank" rel="noopener" href="https://matinnuhamunada.github.io/posts/2022/01/rnammer">How to Install RNAmmer in Prokka - Matin Nuhamunada</a>、<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/ba28e4674961">本地rnammer-1.2使用 - 简书 (jianshu.com)</a>和<a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=1979">RNAmmer的安装和使用 | 陈连福的生信博客 (chenlianfu.com)</a></p>
<h6 id="2-4-3-利用tRNAscan-SE注释tRNA"><a href="#2-4-3-利用tRNAscan-SE注释tRNA" class="headerlink" title="2.4.3 利用tRNAscan-SE注释tRNA"></a>2.4.3 利用tRNAscan-SE注释tRNA</h6><p>利用conda安装</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">conda install trnascan-se=2.0.12</span><br><span class="line"></span><br><span class="line"><span class="comment">#基础用法</span></span><br><span class="line">Basic Options</span><br><span class="line">  -E         : search <span class="keyword">for</span> eukaryotic tRNAs (default)</span><br><span class="line">  -B         : search <span class="keyword">for</span> bacterial tRNAs</span><br><span class="line">  -A         : search <span class="keyword">for</span> archaeal tRNAs</span><br><span class="line">  -M &lt;model&gt; : search <span class="keyword">for</span> mitochondrial tRNAs</span><br><span class="line">                 options: mammal, vert</span><br><span class="line">  -O         : search <span class="keyword">for</span> other organellar tRNAs</span><br><span class="line">  -G         : use general tRNA model (cytoslic tRNAs from all 3 domains included)</span><br><span class="line">  -L         : search using the legacy method (tRNAscan, EufindtRNA, and COVE)</span><br><span class="line">                 use with -E, -B, -A, -O, or -G</span><br><span class="line">  -I         : search using Infernal (default)</span><br><span class="line">                 use with -E, -B, -A, -O, or -G</span><br><span class="line">  -o &lt;file&gt;  : save final results <span class="keyword">in</span> &lt;file&gt;</span><br><span class="line">  -f &lt;file&gt;  : save tRNA secondary structures to &lt;file&gt;</span><br><span class="line">  -m &lt;file&gt;  : save statistics summary <span class="keyword">for</span> run <span class="keyword">in</span> &lt;file&gt;</span><br><span class="line">               (speed, <span class="comment"># tRNAs found in each part of search, etc)</span></span><br><span class="line">  -H         : show both primary and secondary structure components to</span><br><span class="line">               covariance model bit scores</span><br><span class="line">  -q         : quiet mode (credits &amp; run option selections suppressed)</span><br><span class="line"></span><br><span class="line">  -h         : <span class="built_in">print</span> full list (long) of available options</span><br><span class="line">  </span><br><span class="line"><span class="comment">#运行软件</span></span><br><span class="line">tRNAscan-SE -E -o tRNA.out -f tRNA.ss -m tRNA.stats ../purged.fa.masked</span><br></pre></td></tr></table></figure>



<h4 id="3-编码基因注释"><a href="#3-编码基因注释" class="headerlink" title="3 编码基因注释"></a>3 编码基因注释</h4><h5 id="3-1-基因结构预测的方案"><a href="#3-1-基因结构预测的方案" class="headerlink" title="3.1 基因结构预测的方案"></a>3.1 基因结构预测的方案</h5><p>一般包括三部分：①利用近缘物种的蛋白进行同源注释；②利用转录组或isoform测序数据进行基因模型预测；③通过训练已有的数据来从头预测基因组的基因结构。</p>
<h5 id="3-2-参考文章示例"><a href="#3-2-参考文章示例" class="headerlink" title="3.2 参考文章示例"></a>3.2 参考文章示例</h5><p>①A draft genome assembly of reef-building octocoral Heliopora coerulea (Ip, et al., 2023)</p>
<img src="/2024/07/17/genome-anno/image-20240328220942281.png" class title="image-20240328220942281">

<p>②Penaeid shrimp genome provides insights into benthic adaptation and frequent molting (Zhang, et al., 2019)</p>
<img src="/2024/07/17/genome-anno/image-20240328221247850.png" class title="image-20240328221247850">

<img src="/2024/07/17/genome-anno/image-20240328221258675.png" class title="image-20240328221258675">

<p>③Chromosome-level genome assembly of the silver pomfret Pampus argenteus (Wei, et al., 2024)</p>
<img src="/2024/07/17/genome-anno/image-20240328222148370.png" class title="image-20240328222148370">

<h5 id="3-3-基因结构预测的软件"><a href="#3-3-基因结构预测的软件" class="headerlink" title="3.3 基因结构预测的软件"></a>3.3 基因结构预测的软件</h5><p>这部分内容在<a target="_blank" rel="noopener" href="https://www.jianshu.com/u/740f4b0f11e9">Mr_我爱读文献</a>的简书中<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/b75b8e253552">基因组结构注释软件列表 - 简书 (jianshu.com)</a>进行了详细罗列，不再赘述。</p>
<h5 id="3-4-软件安装及其使用"><a href="#3-4-软件安装及其使用" class="headerlink" title="3.4 软件安装及其使用"></a>3.4 软件安装及其使用</h5><h6 id="3-4-1基于转录组数据注释"><a href="#3-4-1基于转录组数据注释" class="headerlink" title="3.4.1基于转录组数据注释"></a>3.4.1基于转录组数据注释</h6><p>策略一：HISAT2 + StringTie + TransDecoder</p>
<p>参考<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/659049169">基因结构预测 |TransDecoder和PASA基于转录组预测及转录本去冗余 - 知乎 (zhihu.com)</a>和<a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/11/24/2020-11-24-annotion(3)/">基因结构注释（3）：转录组注释 - 生信学习 | Zhou Xiaozhao &#x3D; 小钊の笔记 &#x3D; 前天是小兔子，昨天是小鹿，今天是你</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#所需软件可以用conda安装</span></span><br><span class="line">conda install fastp=0.23.4</span><br><span class="line">conda install hisat2=2.2.1</span><br><span class="line">conda install samtools=1.19.2</span><br><span class="line">conda install stringtie=2.2.1</span><br><span class="line"></span><br><span class="line"><span class="comment">#对二代转录组数据进行质控</span></span><br><span class="line">fastp -i Unknown_BD751-01T0002_good_1.fq.gz -o output.R1.fq -I Unknown_BD751-01T0002_good_2.fq.gz -O output.R2.fq -w 8</span><br><span class="line"></span><br><span class="line"><span class="comment">#为屏蔽重复序列后的基因组建立索引</span></span><br><span class="line">hisat2-build ../purged.fa.masked z189.genome.index</span><br><span class="line"></span><br><span class="line"><span class="comment">#将双端的reads进行基因组mapping</span></span><br><span class="line">$ hisat2 -p 20 --dta --no-mixed -x z189.genome.index -1 output.R1.fq -2 output.R2.fq --no-unal -S z189.sam 2&gt;z189.summary.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用samtools对比对结果进行二进制转换和排序</span></span><br><span class="line">samtools view -b z189.sam -o z189.bam</span><br><span class="line">samtools sort z189.bam -o z189.sort.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用stringtie进行转录本的预测</span></span><br><span class="line">stringtie -p 20 -o 02out.gtf z189.sort.bam</span><br></pre></td></tr></table></figure>

<p>最后利用transdecoder注释蛋白编码区域，参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/fdd547223ed5">使用TransDecoder寻找转录本中的编码区 - 简书 (jianshu.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">conda install bioconda::transdecoder=5.5.0</span><br><span class="line">conda install bioconda::diamond=2.1.9</span><br><span class="line"></span><br><span class="line"><span class="comment">#从GTF文件中提取FASTA序列</span></span><br><span class="line">gtf_genome_to_cdna_fasta.pl 02out.gtf ../purged.fa.masked &gt; transcripts.fasta</span><br><span class="line"><span class="comment">#将GTF文件转成GFF3格式</span></span><br><span class="line">gtf_to_alignment_gff3.pl 02out.gtf &gt; 02out.gff3</span><br><span class="line"><span class="comment">#预测转录本中长的开放阅读框</span></span><br><span class="line">TransDecoder.LongOrfs -t transcripts.fasta</span><br><span class="line"></span><br><span class="line"><span class="comment">#对于得到的开放阅读框(以起始密码子为头，以终止密码子为尾)，与uniprot数据库中的蛋白作比较，来检验预测的可靠性，寻找同源证据。这里要下载uniport蛋白库</span></span><br><span class="line">wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz</span><br><span class="line">gunzip uniprot_sprot.fasta.gz</span><br><span class="line"><span class="comment">#在进行blast比对之前，要对蛋白文件进行建库</span></span><br><span class="line">diamond makedb --<span class="keyword">in</span> uniprot_sprot.fasta --db uniprot_sprot.fasta</span><br><span class="line"><span class="comment">#进行比对,diamond blast速度非常快</span></span><br><span class="line">diamond blastp -d uniprot_sprot.fasta -q transcripts.fasta.transdecoder_dir/longest_orfs.pep --evalue 1e-5 --max-target-seqs 1 &gt; blastp.outfmt6</span><br><span class="line"></span><br><span class="line"><span class="comment">#预测可能的编码区</span></span><br><span class="line">TransDecoder.Predict -t transcripts.fasta --retain_blastp_hits blastp.outfmt6</span><br><span class="line"><span class="comment">#生成基于参考基因组的编码区注释文件</span></span><br><span class="line">cdna_alignment_orf_to_genome_orf.pl transcripts.fasta.transdecoder.gff3 02out.gff3 transcripts.fasta &gt; transcripts.fasta.transdecoder.genome.gff3</span><br></pre></td></tr></table></figure>

<p>得到基于转录组数据注释的gff文件transcripts.fasta.transdecoder.genome.gff3</p>
<p>注：没有脚本将该文件转换为evm格式的gff文件，在整合gff文件时的策略是将其和从头注释的gff文件合并，在权重文件中标注OTHER_PREDICTION transdecoder</p>
<p>策略二（可注释不同转录本）：trinity+pasa</p>
<p>首先利用trinity进行转录组的de novo组装</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">sh trinity.sh  (trinity.sh内容如下)</span><br><span class="line"></span><br><span class="line"><span class="comment">## my_trinity_script.sh</span></span><br><span class="line"><span class="comment">#!/bin/bash</span></span><br><span class="line">Trinity --seqType fq \</span><br><span class="line">--left Unknown_BD751-01T0002_good_1.fq.gz \</span><br><span class="line">--right Unknown_BD751-01T0002_good_2.fq.gz \</span><br><span class="line">--CPU  20 \</span><br><span class="line">--max_memory 30G</span><br></pre></td></tr></table></figure>



<h6 id="3-4-2-基于近缘物种的同源蛋白注释"><a href="#3-4-2-基于近缘物种的同源蛋白注释" class="headerlink" title="3.4.2 基于近缘物种的同源蛋白注释"></a>3.4.2 基于近缘物种的同源蛋白注释</h6><p>①利用miniprot软件，下载并提取近缘物种的蛋白序列，看文献中一般用5个物种左右。将蛋白map到基因组，得到gff文件，参考大神李恒的官方描述<a target="_blank" rel="noopener" href="https://github.com/lh3/miniprot?tab=readme-ov-file#usage">lh3&#x2F;miniprot: Align proteins to genomes with splicing and frameshift (github.com)</a>，以下是recommended的方法</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#利用conda安装miniprot</span></span><br><span class="line">conda install miniprot=0.13</span><br><span class="line"></span><br><span class="line"><span class="comment">#由于 miniprot 索引速度较慢且占用大量内存，因此建议预先生成索引</span></span><br><span class="line">miniprot -t8 -d ref.mpi ../purged.fa.masked</span><br><span class="line">miniprot -t8 -I --gff ref.mpi protein.faa &gt; out.gff3</span><br></pre></td></tr></table></figure>

<p>得到基于近缘物种的同源蛋白注释gff文件out.gff3</p>
<p>利用脚本将该文件转换为evm格式的gff文件，脚本地址下载地址<a target="_blank" rel="noopener" href="https://github.com/EVidenceModeler/EVidenceModeler/blob/master/EvmUtils/misc/miniprot_GFF_2_EVM_GFF3.py">EVidenceModeler&#x2F;EvmUtils&#x2F;misc&#x2F;miniprot_GFF_2_EVM_GFF3.py at master · EVidenceModeler&#x2F;EVidenceModeler (github.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python ~/scripts/miniprot_GFF_2_EVM_GFF3.py out.gff3 &gt; out_evm.gff3</span><br></pre></td></tr></table></figure>

<p>②利用gemoma软件，需要下载近缘物种的基因组组装序列文件(fasta格式)、gff3文件和蛋白序列文件，同时也需要上面得到的转录组reads的mapping文件(sam格式)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#利用conda安装，最好新建环境</span></span><br><span class="line">conda create -n gemoma python=3</span><br><span class="line">conda activate gemoma</span><br><span class="line">conda install bioconda::gemoma=1.9</span><br></pre></td></tr></table></figure>

<p>该软件具有多种运行方式：</p>
<p>①GeMoMa GeMoMaPipeline 加参数格式，参考<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/688384179">GeMoMa：基因同源预测软件 - 知乎 (zhihu.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">GeMoMa GeMoMaPipeline threads=10 outdir=`<span class="built_in">pwd</span>` GeMoMa.Score=ReAlign AnnotationFinalizer.r=NO o=<span class="literal">true</span> t=scaffold.fa a=Genome1.gff g=Genome1.fa</span><br></pre></td></tr></table></figure>

<p>②利用java命令运行，参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/8d795097d859">基因注释工具GeMoMa - 简书 (jianshu.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">java -jar GeMoMa-1.9.jar CLI GeMoMaPipeline threads=4 outdir=output GeMoMa.Score=ReAlign AnnotationFinalizer.r=NO o=<span class="literal">true</span> t=target.fa  a=ref.gff g=ref.fa</span><br></pre></td></tr></table></figure>

<p>前两种方法一直报错，再尝试第三种方法</p>
<p>③找到pipeline.sh文件的绝对路径，比如&#x2F;public&#x2F;home&#x2F;wutianzhen&#x2F;miniconda3&#x2F;pkgs&#x2F;gemoma-1.9-hdfd78af_0&#x2F;share&#x2F;gemoma-1.9-0&#x2F;pipeline.sh，再运行该sh文件，参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6d9d9f0c38a6">gemoma安装与简单用法 - 简书 (jianshu.com)</a>。pipeline.sh内容如下，若有转录组数据则设置8个参数，若没有转录组数据则设置前6个参数。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line">jar=$(<span class="built_in">eval</span> <span class="string">&quot;ls /public/home/wutianzhen/miniconda3/pkgs/gemoma-1.9-hdfd78af_0/share/gemoma-1.9-0/GeMoMa-*.jar&quot;</span>)  <span class="comment">#此处ls后面最好添加GeMoMa-1.9.jar的绝对路径，否则只能在GeMoMa-1.9.jar的目录下运行程序</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># This script allows to run the complete GeMoMa pipeline multithreaded from the command line.</span></span><br><span class="line"><span class="comment"># The final prediction is located in $&#123;out&#125;/filtered_predictions.gff.</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment"># A simple example without RNA-seq using tblastn is</span></span><br><span class="line"><span class="comment"># ./pipeline.sh tblastn tests/gemoma/target-fragment.fasta tests/gemoma/ref-annotation.gff tests/gemoma/ref-fragment.fasta 1 results/sw-pipeline</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#parameters</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="string">&quot;<span class="variable">$1</span>&quot;</span> == <span class="string">&quot;tblastn&quot;</span> ]  <span class="comment">#设置比对方法为tblastn</span></span><br><span class="line"><span class="keyword">then</span></span><br><span class="line">        tblastn=<span class="literal">true</span>;</span><br><span class="line"><span class="keyword">else</span></span><br><span class="line">        tblastn=<span class="literal">false</span>;</span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line">target_genome=<span class="variable">$2</span>          <span class="comment">#设置我们自己的基因组文件，后缀必须是(fasta,fa,fas,fna,fasta.gz,fa.gz,fas.gz,fna.gz)其中的一种</span></span><br><span class="line">ref_annotation=<span class="variable">$3</span>         <span class="comment">#设置参考基因组注释文件</span></span><br><span class="line">ref_genome=<span class="variable">$4</span>             <span class="comment">#设置参考基因组序列文件</span></span><br><span class="line">threads=<span class="variable">$5</span>                <span class="comment">#设置运行所用的核心数</span></span><br><span class="line">out=<span class="variable">$6</span>                    <span class="comment">#设置输出路径</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$#</span> -ne 6 ]; <span class="keyword">then</span></span><br><span class="line">        lib=<span class="variable">$7</span>;           <span class="comment">#设置转录组数据的类型，其中FR_UNSTRANDED表示没方向，FR_FIRST_STRAND定向转录的正义链，FR_SECOND_STRAND定向转录的反义链</span></span><br><span class="line">        reads=<span class="variable">$8</span>;         <span class="comment">#转录组mapping的sam或bam文件</span></span><br><span class="line">        <span class="built_in">echo</span> <span class="string">&quot;GeMoMa using RNA-seq data: library type=&quot;</span> <span class="variable">$lib</span> <span class="string">&quot;mapped reads=&quot;</span> <span class="variable">$reads</span></span><br><span class="line">        time java -jar <span class="variable">$jar</span> CLI GeMoMaPipeline threads=<span class="variable">$threads</span> t=<span class="variable">$target_genome</span> s=own a=<span class="variable">$ref_annotation</span> g=<span class="variable">$ref_genome</span> tblastn=<span class="variable">$&#123;tblastn&#125;</span> outdir=<span class="variable">$out</span> r=MAPPED ERE.s=<span class="variable">$lib</span> ERE.m=<span class="variable">$reads</span> ERE.c=<span class="literal">true</span> AnnotationFinalizer.r=NO</span><br><span class="line"><span class="keyword">else</span></span><br><span class="line">        <span class="built_in">echo</span> <span class="string">&quot;GeMoMa without RNA-seq data&quot;</span></span><br><span class="line">        time java -jar <span class="variable">$jar</span> CLI GeMoMaPipeline threads=<span class="variable">$threads</span> t=<span class="variable">$target_genome</span> s=own a=<span class="variable">$ref_annotation</span> g=<span class="variable">$ref_genome</span> tblastn=<span class="variable">$&#123;tblastn&#125;</span> outdir=<span class="variable">$out</span> AnnotationFinalizer.r=NO</span><br><span class="line"><span class="keyword">fi</span></span><br></pre></td></tr></table></figure>

<p>运行sh脚本</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sh /public/home/wutianzhen/miniconda3/pkgs/gemoma-1.9-hdfd78af_0/share/gemoma-1.9-0/pipeline.sh tblastn purged.fa.masked.fas genomic.gff dgig.fna 5 ./result FR_UNSTRANDED ../../02Transcriptome/z189.bam</span><br></pre></td></tr></table></figure>

<p>第三种方法运行成功，仍遇到了两个报错，一是GeMoMa-1.9.jar的路径问题，而是target基因组的后缀问题。前两种方法的报错可能也是由于这些问题。</p>
<p>GeMoMa运行结束产生final_annotation.gff文件。</p>
<p>利用脚本将该文件转换为evm格式的gff文件，脚本地址<a target="_blank" rel="noopener" href="https://github.com/EVidenceModeler/EVidenceModeler/blob/master/EvmUtils/misc/GeMoMa_gff_to_gff3.pl">EVidenceModeler&#x2F;EvmUtils&#x2F;misc&#x2F;GeMoMa_gff_to_gff3.pl at master · EVidenceModeler&#x2F;EVidenceModeler (github.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl ~/scripts/GeMoMa_gff_to_gff3.pl final_annotation.gff &gt; final_annotation_evm.gff</span><br></pre></td></tr></table></figure>

<p>脚本运行时报错缺少三个模块：Gene_obj.pm、Nuc_translator.pm和Longest_orf.pm，</p>
<p>这些模块可以在<a target="_blank" rel="noopener" href="https://github.com/EVidenceModeler/EVidenceModeler/blob/master/PerlLib/Longest_orf.pm">EVidenceModeler&#x2F;PerlLib&#x2F;Longest_orf.pm at master · EVidenceModeler&#x2F;EVidenceModeler (github.com)</a>下载得到</p>
<p>将它们放在已有的@INC目录中，这个目录通过以下命令来查询</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl -e &#x27;&#123;print &quot;$_\n&quot; foreach @INC&#125;&#x27;</span><br></pre></td></tr></table></figure>

<img src="/2024/07/17/genome-anno/image-20240419111329510.png" class title="image-20240419111329510">

<p>将下载的perl模块放在&#x2F;public&#x2F;home&#x2F;wutianzhen&#x2F;miniconda3&#x2F;envs&#x2F;annotation&#x2F;lib&#x2F;perl5&#x2F;core_perl&#x2F;下面即可</p>
<h6 id="3-4-3-从头基因结构预测"><a href="#3-4-3-从头基因结构预测" class="headerlink" title="3.4.3 从头基因结构预测"></a>3.4.3 从头基因结构预测</h6><p>利用Augustus进行基因模型训练，训练数据可以下载近缘物种的基因组数据和gff3文件。</p>
<p>参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6f7b2998600c">Augustus 基因从头预测 - 简书 (jianshu.com)</a>和<a target="_blank" rel="noopener" href="https://docs.hpc.sjtu.edu.cn/app/bioinformatics/augustus.html">AUGUSTUS - 上海交大超算平台用户手册 Documentation (sjtu.edu.cn)</a></p>
<p>建议重建一个conda环境再进行安装</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#安装用到的软件</span></span><br><span class="line">conda install bioconda::augustus=3.5</span><br><span class="line">conda install gffread=0.12.7</span><br><span class="line">conda install seqkit=2.8.1</span><br><span class="line"></span><br><span class="line"><span class="comment">#首先得到蛋白序列</span></span><br><span class="line">gffread genomic.gff -g GCF_004324835.1_DenGig_1.0_genomic.fna -y dgig.pep</span><br><span class="line"></span><br><span class="line"><span class="comment">#得到genebank格式文件</span></span><br><span class="line">gff2gbSmallDNA.pl genomic.gff GCF_004324835.1_DenGig_1.0_genomic.fna 1000 gene.raw.gb</span><br><span class="line"></span><br><span class="line"><span class="comment">#尝试训练，捕捉错误基因</span></span><br><span class="line">etraining --species=generic --stopCodonExcludedFromCDS=<span class="literal">false</span> gene.raw.gb 2&gt; train.err</span><br><span class="line"></span><br><span class="line"><span class="comment">#过滤掉错误的基因结构</span></span><br><span class="line">cat train.err | perl -pe <span class="string">&#x27;s/.*in sequence (\S+): .*/$1/&#x27;</span> &gt;badgenes.lst</span><br><span class="line">filterGenes.pl badgenes.lst gene.raw.gb &gt; genes.gb</span><br><span class="line"></span><br><span class="line"><span class="comment">#提取过滤后的蛋白</span></span><br><span class="line">grep <span class="string">&#x27;/gene&#x27;</span> genes.gb |sort |uniq  |sed <span class="string">&#x27;s/\/gene=//g&#x27;</span> |sed <span class="string">&#x27;s/\&quot;//g&#x27;</span> |awk <span class="string">&#x27;&#123;print $1&#125;&#x27;</span> &gt;geneSet.lst</span><br><span class="line">seqkit grep -f geneSet.lst dgig.pep &gt;geneSet.lst.fa   <span class="comment">#19708个蛋白模型</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#将得到的蛋白序列进行建库，自身blastp比对。根据比对结果，如果基因间identity &gt;= 70%，则只保留其中之一，再次得到一个过滤后的gff文件，gene_filter.gff3</span></span><br><span class="line">makeblastdb -<span class="keyword">in</span> geneSet.lst.fa -dbtype prot -parse_seqids -out geneSet.lst.fa</span><br><span class="line">blastp -db geneSet.lst.fa -query geneSet.lst.fa -out geneSet.lst.fa.blastp -evalue 1e-5 -outfmt 6 -num_threads 8</span><br><span class="line"></span><br><span class="line">awk <span class="string">&#x27;$3 &gt; 70 &amp;&amp; $1 != $2 &#123;print $2&#125;&#x27;</span> geneSet.lst.fa.blastp | sort | uniq &gt; filtered_lines.txt</span><br><span class="line">awk <span class="string">&#x27;NR==FNR&#123;a[$0];next&#125; !($0 in a)&#x27;</span> filtered_lines.txt genomic.gff &gt; gene_filter.gff3</span><br><span class="line"></span><br><span class="line"><span class="comment">#将得到的gene_filter.gff3 转换为genbank 格式文件</span></span><br><span class="line">gff2gbSmallDNA.pl gene_filter.gff3 GCF_004324835.1_DenGig_1.0_genomic.fna 1000 genes.gb.filter</span><br><span class="line"></span><br><span class="line"><span class="comment">#将上一步过滤后的文件随机分成两份，测试集和训练集，100为测试集的基因数目，其余为训练集</span></span><br><span class="line">randomSplit.pl genes.gb.filter 100</span><br><span class="line"></span><br><span class="line"><span class="comment">#进行训练</span></span><br><span class="line">new_species.pl --species=z189</span><br><span class="line">etraining --species=z189 genes.gb.filter.train</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用测试集进行检验</span></span><br><span class="line">augustus --species=z189 genes.gb.filter.test | tee firsttest.out</span><br><span class="line">augustus --species=nematostella_vectensis genes.gb.filter.test | tee firsttest_nvec.out</span><br><span class="line">augustus --species=human genes.gb.filter.test | tee firsttest_human.out</span><br></pre></td></tr></table></figure>

<p>测试结果如下</p>
<img src="/2024/07/17/genome-anno/image-20240412151218427.png" class title="image-20240412151218427">

<p>近缘物种nematostella_vectensis训练情况的查看</p>
<img src="/2024/07/17/genome-anno/image-20240412151807169.png" class title="image-20240412151807169">

<p>人类human的训练情况如下</p>
<img src="/2024/07/17/genome-anno/image-20240412152306450.png" class title="image-20240412152306450">

<p>通过比较发现，训练效果优于通过近缘已训练物种的预测，但是gene level的sensitivity只有37%，不知道能不能继续下一步，还是优化训练。</p>
<p>其他推文说低于20%的话需要加大数据量继续优化训练，并且优化仅能提高几个百分点，因此这里初步训练的结果应该是可以拿来从头基因注释的。</p>
<p>接下来进行进行CRF训练，CRF： conditional random field， 进行进行CRF时，将备份&#x2F;species&#x2F;target species&#x2F;中的所有参数。比较CRF前后预测的精确性，若升高则使用，若降低，则用上一步结果</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">etraining --species=z189 --CRF=1 genes.gb.filter.train</span><br><span class="line">augustus --species=z189 genes.gb.filter.test | tee secondtest.out.withCRF</span><br></pre></td></tr></table></figure>

<p>进行基因结构预测，得到最后的gff文件augustus_z189_gene.gff</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">augustus --species=z189 --gff3=on purged.fa.masked &gt;augustus_z189.gff <span class="comment">#使用默认参数</span></span><br><span class="line"></span><br><span class="line">perl ~/scripts/augustus_gff3_to_evm_gff3.pl augustus_z189.gff &gt; augustus_z189_gene.gff <span class="comment">#使用evm脚本将gff转化为evm格式的gff，脚本下载链接：https://github.com/Zhanmengtao/bin/blob/master/augustus_gff3_to_evm_gff3.pl</span></span><br></pre></td></tr></table></figure>

<h6 id="3-4-5-将以上不同软件的注释结果进行整合"><a href="#3-4-5-将以上不同软件的注释结果进行整合" class="headerlink" title="3.4.5 将以上不同软件的注释结果进行整合"></a>3.4.5 将以上不同软件的注释结果进行整合</h6><p>参考<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/ca4ac8700e32">EVM 对预测结果进行整合 - 简书 (jianshu.com)</a></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget -c https://github.com/EVidenceModeler/EVidenceModeler/archive/refs/heads/master.zip</span><br><span class="line">unzip master.zip</span><br><span class="line">cd EVidenceModeler-master/</span><br><span class="line">EVidenceModeler -h</span><br></pre></td></tr></table></figure>

<p>准备好三种预测的结果文件transcript_alignments.gff3，protein_alignments.gff3和gene_prediction.gff3</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl -p -i -e <span class="string">&#x27;s/^#.*//s&#x27;</span> gene_prediction.gff3 transcript_alignments.gff3 protein_alignments.gff3</span><br></pre></td></tr></table></figure>

<p>编辑注释证据的权重文件</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">vi weights.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">#内容如下</span></span><br><span class="line"></span><br><span class="line">PROTEIN miniprot        2</span><br><span class="line">PROTEIN GAF     1</span><br><span class="line">PROTEIN GeMoMa  1</span><br><span class="line">TRANSCRIPT      transdecoder    8</span><br><span class="line">ABINITIO_PREDICTION     AUGUSTUS        1</span><br></pre></td></tr></table></figure>

<p>分步运行程序</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#将从头注释和转录组注释结果合并</span></span><br><span class="line">cat transcript_alignments.gff3 gene_prediction.gff3 &gt; gene_prediction_combined.gff3</span><br><span class="line"></span><br><span class="line"><span class="comment">#对每一个contig序列进行单独创建文件夹，将gff3文件划分开来</span></span><br><span class="line">perl EVidenceModeler-master/EvmUtils/partition_EVM_inputs.pl --partition_dir ./partition --genome ../purged.fa.masked --gene_predictions gene_prediction_combined.gff3 --protein_alignments protein_alignments.gff3 --segmentSize 100000 --overlapSize 10000 --partition_listing partitions_list.out</span><br><span class="line"></span><br><span class="line"><span class="comment">#生成批量运行evm的sh文件</span></span><br><span class="line">perl EVidenceModeler-master/EvmUtils/write_EVM_commands.pl --partitions partitions_list.out --genome ../purged.fa.masked --gene_predictions gene_prediction_combined.gff3 --protein_alignments protein_alignments.gff3 --output_file_name evm.out --weights `<span class="built_in">pwd</span>`/weights.txt &gt; commands.list<span class="string">&quot;</span></span><br><span class="line"><span class="string">      </span></span><br><span class="line"><span class="string">#运行evm程序</span></span><br><span class="line"><span class="string">perl EVidenceModeler-master/EvmUtils/execute_EVM_commands.pl commands.list | tee evm_run.log</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">#合并运行结果</span></span><br><span class="line"><span class="string">perl EVidenceModeler-master/EvmUtils/recombine_EVM_partial_outputs.pl --partitions partitions_list.out --output_file_name evm.out</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">#结果整理为gff3格式</span></span><br><span class="line"><span class="string">perl EVidenceModeler-master/EvmUtils/convert_EVM_outputs_to_GFF3.pl --partitions partitions_list.out --output_file_name evm.out --genome ../purged.fa.masked</span></span><br><span class="line"><span class="string">find . -regex &quot;</span>.*evm.out.gff3<span class="string">&quot; -exec cat &#123;&#125; \; | bedtools sort -i - &gt; EVM.all.gff</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">#过滤gff文件,过滤出长度不足50AA的序列</span></span><br><span class="line"><span class="string">conda install gffread</span></span><br><span class="line"><span class="string">conda install bioconda::bioawk</span></span><br><span class="line"><span class="string">gffread EVM.all.gff -g ../purged.fa.masked -y tr_cds.fa</span></span><br><span class="line"><span class="string">bioawk -c fastx &#x27;length(<span class="variable">$seq</span>) &lt; 50 &#123;print <span class="variable">$name</span>&#125;&#x27; tr_cds.fa | sed &#x27;s/evm.model.//g&#x27; &gt; short_aa_gene_list.txt #这一步骤利用别人的推送中的命令跑不通，进行了修改</span></span><br><span class="line"><span class="string">grep -v -w -f short_aa_gene_list.txt EVM.all.gff &gt; z189.gff</span></span><br></pre></td></tr></table></figure>

<p>到这里，得到了初步地结构注释结果文件z189.gff</p>
<h6 id="3-5-busco评估基因注释质量"><a href="#3-5-busco评估基因注释质量" class="headerlink" title="3.5 busco评估基因注释质量"></a>3.5 busco评估基因注释质量</h6><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#利用真核生物库</span></span><br><span class="line">busco_for_annotation]$ busco -i tr_cds.fa -l /project/tianzhenWu/z189/02genome_assemble/without-hic/fasta_asm/nextpolish/01_rundir/eukaryota_odb10/ -m prot -o busco_out -c 40</span><br><span class="line"></span><br><span class="line"><span class="comment">#利用后生动物库</span></span><br><span class="line">busco -i tr_cds.fa -l /project/tianzhenWu/z189/02genome_assemble/without-hic/fasta_asm/nextpolish/01_rundir/metazoa_odb10/ -m prot -o busco_out_m -c 40</span><br></pre></td></tr></table></figure>

<p>利用真核生物库评估结果如下：</p>
<img src="/2024/07/17/genome-anno/image-20240419202621349.png" class title="image-20240419202621349">

<h6 id="3-6-对基因特征进行可视化，包括基因长度、基因间长度、外显子个数等"><a href="#3-6-对基因特征进行可视化，包括基因长度、基因间长度、外显子个数等" class="headerlink" title="3.6 对基因特征进行可视化，包括基因长度、基因间长度、外显子个数等"></a>3.6 对基因特征进行可视化，包括基因长度、基因间长度、外显子个数等</h6><p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/614002824">玩转基因组 | 利用R语言统计外显子数量和基因长度并可视化 - 知乎 (zhihu.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/323612a6ed0a">统计注释出的基因数目，基因长度，外显子和CDS长度 - 简书 (jianshu.com)</a></p>
<p>#将gff转换为gtf文件</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gffread z189.gff -T -o z189.gtf</span><br></pre></td></tr></table></figure>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">\<span class="comment">#统计基因模型总长度、数目及平均基因长度</span></span><br><span class="line"></span><br><span class="line">cat maker_rnd3.gff | awk <span class="string">&#x27;&#123; if ($3 == &quot;gene&quot;) print $0 &#125;&#x27;</span> | awk <span class="string">&#x27;&#123; sum += ($5 - $4) &#125; END &#123; print sum, NR, sum / NR &#125;&#x27;</span></span><br><span class="line"></span><br><span class="line">308328514 42256 7296.68</span><br><span class="line"></span><br><span class="line"> \<span class="comment">#统计外显子总长度、数目以及平均外线子长度</span></span><br><span class="line"></span><br><span class="line">cat z189.gff | awk <span class="string">&#x27;&#123; if ($3 == &quot;exon&quot;) print $0 &#125;&#x27;</span> | awk <span class="string">&#x27;&#123; sum += ($5 - $4) &#125; END &#123; print sum, NR, sum / NR &#125;&#x27;</span></span><br><span class="line"></span><br><span class="line">50327093 198915 253.008</span><br></pre></td></tr></table></figure>



<h4 id="4-基因功能注释"><a href="#4-基因功能注释" class="headerlink" title="4 基因功能注释"></a>4 基因功能注释</h4><p>常用数据据库:NR，Uniprot (Swiss-Prot, TrEMBL), eggNOG, KOG, KEGG, Go,Pfam等，参考<a target="_blank" rel="noopener" href="https://phantom-aria.github.io/2023/09/19/a.html#1-5-Swissprot%E6%95%B0%E6%8D%AE%E5%BA%93">基因组注释（6）——在线版eggNOG-mapper注释功能基因 - 我的小破站 (phantom-aria.github.io)</a></p>
<h5 id="4-1-利用NR动物数据库注释"><a href="#4-1-利用NR动物数据库注释" class="headerlink" title="4.1 利用NR动物数据库注释"></a>4.1 利用NR动物数据库注释</h5><p>下载nr数据库，参考<a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/207/">创建NR子库以及从NR库提取特定物种分类的序列 | KeepNotes blog (bioinfo-scrounger.com)</a>和<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/f78a98587f8c">2022-09-03 NR 动物数据库构建方法2 - 简书 (jianshu.com)</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda install -y -c hcc aspera-cli</span><br><span class="line">conda install -y -c bioconda sra-tools</span><br><span class="line">ascp -v -k 1 -T -l 200m -i /public/home/wutianzhen/miniconda3/envs/download/etc/asperaweb_id_dsa.openssh anonftp@ftp.ncbi.nlm.nih.gov:/blast/db/FASTA/nr.gz ./    <span class="comment">#需要找到密钥文件asperaweb_id_dsa.openssh的绝对路径</span></span><br></pre></td></tr></table></figure>

<p>下载其他分类相关库，然后提取6073 刺胞动物子库</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda taxonkit</span><br><span class="line">conda install -c bioconda csvtk</span><br><span class="line">wget -c ftp://ftp.ncbi.nih.gov/pub/taxonomy/accession2taxid/prot.accession2taxid.gz</span><br><span class="line">wget -c https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz</span><br><span class="line">tar -zxvf taxdump.tar.gz</span><br><span class="line">cp names.dmp ~/.taxonkit</span><br><span class="line">cp nodes.dmp ~/.taxonkit</span><br><span class="line">taxonkit list --ids 6073 --indent <span class="string">&quot;&quot;</span> &gt; cnidaria.taxid.txt</span><br><span class="line">wc -l cnidaria.taxid.txt  <span class="comment">##打印15570 cnidaria.taxid.txt</span></span><br><span class="line">zcat prot.accession2taxid.gz |csvtk -t grep -f taxid -P cnidaria.taxid.txt |csvtk -t cut -f accession.version &gt; cnidaria.taxid.acc.txt</span><br><span class="line">seqkit grep -f cnidaria.taxid.acc.txt nr -o cnidaria.fas  <span class="comment">#共755344条序列</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#利用diamond进行比对</span></span><br><span class="line">diamond makedb --<span class="keyword">in</span> cnidaria.fas -d cnidaria_diamond.dmnd</span><br><span class="line">diamond blastp --db cnidaria_diamond.dmnd --query z189.pep.fa --out nr.tab --outfmt 6 --sensitive --max-target-seqs 1 --evalue 1e-5 --id 30 --block-size 20 --index-chunks 1</span><br></pre></td></tr></table></figure>

<h5 id="4-2-利用eggNOG数据库进行GO、KEGG、Pfam注释"><a href="#4-2-利用eggNOG数据库进行GO、KEGG、Pfam注释" class="headerlink" title="4.2 利用eggNOG数据库进行GO、KEGG、Pfam注释"></a>4.2 利用eggNOG数据库进行GO、KEGG、Pfam注释</h5><p>网页版注释<a target="_blank" rel="noopener" href="http://eggnog-mapper.embl.de/">eggNOG-mapper (embl.de)</a>，操作简单，修改e值为1e-5</p>
<h5 id="4-3-利用swissprot数据库进行注释"><a href="#4-3-利用swissprot数据库进行注释" class="headerlink" title="4.3 利用swissprot数据库进行注释"></a>4.3 利用swissprot数据库进行注释</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz ./</span><br><span class="line">gzip -d uniprot_sprot.fasta.gz</span><br><span class="line"></span><br><span class="line">diamond makedb --<span class="keyword">in</span> uniprot_sprot.fasta -d uniprot_diamond.dmnd</span><br><span class="line">diamond blastp --db uniprot_diamond.dmnd --query ../05function_annotation/z189.pep.fa --out swissprot.tab --outfmt 6 --sensitive --max-target-seqs 1 --evalue 1e-5 --id 30 --block-size 20 --index-chunks 1</span><br></pre></td></tr></table></figure>

<h5 id="4-4-利用KOG数据库进行注释"><a href="#4-4-利用KOG数据库进行注释" class="headerlink" title="4.4 利用KOG数据库进行注释"></a>4.4 利用KOG数据库进行注释</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">wget https://ftp.ncbi.nih.gov/pub/COG/KOG/fun.txt</span><br><span class="line">wget https://ftp.ncbi.nih.gov/pub/COG/KOG/kog</span><br><span class="line">wget https://ftp.ncbi.nih.gov/pub/COG/KOG/kyva</span><br><span class="line">wget https://ftp.ncbi.nih.gov/pub/COG/KOG/twog</span><br><span class="line"></span><br><span class="line">diamond makedb --<span class="keyword">in</span> kyva.fas -d kog_diamond.dmnd</span><br><span class="line">diamond blastp --db kog_diamond.dmnd --query ../05function_annotation/z189.pep.fa --out kog.tab --outfmt 6 --sensitive --max-target-seqs 1 --evalue 1e-5 --id 30 --block-size 20 --index-chunks 1</span><br></pre></td></tr></table></figure>

<p>利用excel整理结果。</p>
<p>制作VN图：<a target="_blank" rel="noopener" href="https://jvenn.toulouse.inrae.fr/app/example.html">jvenn (inrae.fr)</a>，最多可以做6组。</p>
<link rel="stylesheet" href="/css/spoiler.css" type="text/css"><script src="/js/spoiler.js" type="text/javascript" async></script></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="https://wu-tz.github.io">吴 田 振</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="https://wu-tz.github.io/2024/07/17/genome-anno/">https://wu-tz.github.io/2024/07/17/genome-anno/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="https://wu-tz.github.io" target="_blank">TianzhenWu' Blog</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/genome%E6%B3%A8%E9%87%8A/">genome注释</a></div><div class="post_share"><div class="social-share" data-image="/img/orca-295051_1280.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏</div><div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="/img/wechat.jpg" target="_blank"><img class="post-qr-code-img" src="/img/wechat.jpg" alt="wechat"/></a><div class="post-qr-code-desc">wechat</div></li><li class="reward-item"><a href="/img/alipay.jpg" target="_blank"><img class="post-qr-code-img" src="/img/alipay.jpg" alt="alipay"/></a><div class="post-qr-code-desc">alipay</div></li></ul></div></div><nav class="pagination-post" id="pagination"><div class="next-post pull-full"><a href="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/" title="QuIBL方法检验基因渐渗和不完全谱系分选"><img class="cover" src="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/2024-04-09%20154337.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">QuIBL方法检验基因渐渗和不完全谱系分选</div></div></a></div></nav><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="gitalk-container"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/orca-295051_1280.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">吴 田 振</div><div class="author-info__description">努力发一区</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">29</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">18</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><a id="card-info-btn" href="https://wu-tz.github.io/"><i class="iconfont icon-CSDN"></i><span>收藏我啊</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/Wu-tz" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="mailto:wtzsdut@163.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a><a class="social-icon" href="tencent://AddContact/?fromId=45&amp;fromSubId=1&amp;subcmd=all&amp;uin=2312164805&amp;website=www.oicqzone.com" target="_blank" title="QQ"><i class="fab fa-qq"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">在这里记录的是一些随笔，包括文件处理技巧、软件使用、生物知识等，作为我的工作以及生活记录。欢迎您的访问，祝工作顺利！喜欢的话欢迎收藏~</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%B8%BB%E9%A2%98%EF%BC%9A%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B3%A8%E9%87%8A"><span class="toc-number">1.</span> <span class="toc-text">主题：基因组注释</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-%E9%87%8D%E5%A4%8D%E5%BA%8F%E5%88%97%E6%B3%A8%E9%87%8A"><span class="toc-number">1.1.</span> <span class="toc-text">1 重复序列注释</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#1-1-%E9%87%8D%E5%A4%8D%E5%BA%8F%E5%88%97%E7%9A%84%E4%B8%BB%E8%A6%81%E5%88%86%E7%B1%BB"><span class="toc-number">1.1.1.</span> <span class="toc-text">1.1  重复序列的主要分类</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#1-2-%E6%B3%A8%E9%87%8A%E8%BD%AF%E4%BB%B6"><span class="toc-number">1.1.2.</span> <span class="toc-text">1.2 注释软件</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#1-3-%E7%94%A8%E4%BA%8E%E5%90%8C%E6%BA%90%E6%B3%A8%E9%87%8A%E7%9A%84%E6%95%B0%E6%8D%AE%E5%BA%93"><span class="toc-number">1.1.3.</span> <span class="toc-text">1.3 用于同源注释的数据库</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#1-4-%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E4%B8%8E%E4%BD%BF%E7%94%A8"><span class="toc-number">1.1.4.</span> <span class="toc-text">1.4 软件安装与使用</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-%E9%9D%9E%E7%BC%96%E7%A0%81RNA%E6%B3%A8%E9%87%8A"><span class="toc-number">1.2.</span> <span class="toc-text">2 非编码RNA注释</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#2-1-%E9%9D%9E%E7%BC%96%E7%A0%81rRNA%E5%88%86%E7%B1%BB"><span class="toc-number">1.2.1.</span> <span class="toc-text">2.1 非编码rRNA分类</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#2-2-%E6%B3%A8%E9%87%8A%E8%BD%AF%E4%BB%B6"><span class="toc-number">1.2.2.</span> <span class="toc-text">2.2 注释软件</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#2-3-%E7%94%A8%E4%BA%8E%E6%B3%A8%E9%87%8A%E7%9A%84%E6%95%B0%E6%8D%AE%E5%BA%93"><span class="toc-number">1.2.3.</span> <span class="toc-text">2.3 用于注释的数据库</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#2-4-%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E4%B8%8E%E4%BD%BF%E7%94%A8"><span class="toc-number">1.2.4.</span> <span class="toc-text">2.4 软件安装与使用</span></a><ol class="toc-child"><li class="toc-item toc-level-6"><a class="toc-link" href="#2-4-1-%E5%88%A9%E7%94%A8Infernal%E8%BD%AF%E4%BB%B6%E6%B3%A8%E9%87%8AncRNA"><span class="toc-number">1.2.4.1.</span> <span class="toc-text">2.4.1 利用Infernal软件注释ncRNA</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#2-4-2-%E5%88%A9%E7%94%A8rnammer-%E6%B3%A8%E9%87%8ArRNA"><span class="toc-number">1.2.4.2.</span> <span class="toc-text">2.4.2 利用rnammer 注释rRNA</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#2-4-3-%E5%88%A9%E7%94%A8tRNAscan-SE%E6%B3%A8%E9%87%8AtRNA"><span class="toc-number">1.2.4.3.</span> <span class="toc-text">2.4.3 利用tRNAscan-SE注释tRNA</span></a></li></ol></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-%E7%BC%96%E7%A0%81%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A"><span class="toc-number">1.3.</span> <span class="toc-text">3 编码基因注释</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#3-1-%E5%9F%BA%E5%9B%A0%E7%BB%93%E6%9E%84%E9%A2%84%E6%B5%8B%E7%9A%84%E6%96%B9%E6%A1%88"><span class="toc-number">1.3.1.</span> <span class="toc-text">3.1 基因结构预测的方案</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#3-2-%E5%8F%82%E8%80%83%E6%96%87%E7%AB%A0%E7%A4%BA%E4%BE%8B"><span class="toc-number">1.3.2.</span> <span class="toc-text">3.2 参考文章示例</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#3-3-%E5%9F%BA%E5%9B%A0%E7%BB%93%E6%9E%84%E9%A2%84%E6%B5%8B%E7%9A%84%E8%BD%AF%E4%BB%B6"><span class="toc-number">1.3.3.</span> <span class="toc-text">3.3 基因结构预测的软件</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#3-4-%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85%E5%8F%8A%E5%85%B6%E4%BD%BF%E7%94%A8"><span class="toc-number">1.3.4.</span> <span class="toc-text">3.4 软件安装及其使用</span></a><ol class="toc-child"><li class="toc-item toc-level-6"><a class="toc-link" href="#3-4-1%E5%9F%BA%E4%BA%8E%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E6%B3%A8%E9%87%8A"><span class="toc-number">1.3.4.1.</span> <span class="toc-text">3.4.1基于转录组数据注释</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#3-4-2-%E5%9F%BA%E4%BA%8E%E8%BF%91%E7%BC%98%E7%89%A9%E7%A7%8D%E7%9A%84%E5%90%8C%E6%BA%90%E8%9B%8B%E7%99%BD%E6%B3%A8%E9%87%8A"><span class="toc-number">1.3.4.2.</span> <span class="toc-text">3.4.2 基于近缘物种的同源蛋白注释</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#3-4-3-%E4%BB%8E%E5%A4%B4%E5%9F%BA%E5%9B%A0%E7%BB%93%E6%9E%84%E9%A2%84%E6%B5%8B"><span class="toc-number">1.3.4.3.</span> <span class="toc-text">3.4.3 从头基因结构预测</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#3-4-5-%E5%B0%86%E4%BB%A5%E4%B8%8A%E4%B8%8D%E5%90%8C%E8%BD%AF%E4%BB%B6%E7%9A%84%E6%B3%A8%E9%87%8A%E7%BB%93%E6%9E%9C%E8%BF%9B%E8%A1%8C%E6%95%B4%E5%90%88"><span class="toc-number">1.3.4.4.</span> <span class="toc-text">3.4.5 将以上不同软件的注释结果进行整合</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#3-5-busco%E8%AF%84%E4%BC%B0%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A%E8%B4%A8%E9%87%8F"><span class="toc-number">1.3.4.5.</span> <span class="toc-text">3.5 busco评估基因注释质量</span></a></li><li class="toc-item toc-level-6"><a class="toc-link" href="#3-6-%E5%AF%B9%E5%9F%BA%E5%9B%A0%E7%89%B9%E5%BE%81%E8%BF%9B%E8%A1%8C%E5%8F%AF%E8%A7%86%E5%8C%96%EF%BC%8C%E5%8C%85%E6%8B%AC%E5%9F%BA%E5%9B%A0%E9%95%BF%E5%BA%A6%E3%80%81%E5%9F%BA%E5%9B%A0%E9%97%B4%E9%95%BF%E5%BA%A6%E3%80%81%E5%A4%96%E6%98%BE%E5%AD%90%E4%B8%AA%E6%95%B0%E7%AD%89"><span class="toc-number">1.3.4.6.</span> <span class="toc-text">3.6 对基因特征进行可视化，包括基因长度、基因间长度、外显子个数等</span></a></li></ol></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-%E5%9F%BA%E5%9B%A0%E5%8A%9F%E8%83%BD%E6%B3%A8%E9%87%8A"><span class="toc-number">1.4.</span> <span class="toc-text">4 基因功能注释</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#4-1-%E5%88%A9%E7%94%A8NR%E5%8A%A8%E7%89%A9%E6%95%B0%E6%8D%AE%E5%BA%93%E6%B3%A8%E9%87%8A"><span class="toc-number">1.4.1.</span> <span class="toc-text">4.1 利用NR动物数据库注释</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#4-2-%E5%88%A9%E7%94%A8eggNOG%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9B%E8%A1%8CGO%E3%80%81KEGG%E3%80%81Pfam%E6%B3%A8%E9%87%8A"><span class="toc-number">1.4.2.</span> <span class="toc-text">4.2 利用eggNOG数据库进行GO、KEGG、Pfam注释</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#4-3-%E5%88%A9%E7%94%A8swissprot%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9B%E8%A1%8C%E6%B3%A8%E9%87%8A"><span class="toc-number">1.4.3.</span> <span class="toc-text">4.3 利用swissprot数据库进行注释</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#4-4-%E5%88%A9%E7%94%A8KOG%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9B%E8%A1%8C%E6%B3%A8%E9%87%8A"><span class="toc-number">1.4.4.</span> <span class="toc-text">4.4 利用KOG数据库进行注释</span></a></li></ol></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/07/17/genome-anno/" title="genome_anno">genome_anno</a><time datetime="2024-07-17T13:06:54.000Z" title="发表于 2024-07-17 21:06:54">2024-07-17</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/" title="QuIBL方法检验基因渐渗和不完全谱系分选"><img src="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/2024-04-09%20154337.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="QuIBL方法检验基因渐渗和不完全谱系分选"/></a><div class="content"><a class="title" href="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/" title="QuIBL方法检验基因渐渗和不完全谱系分选">QuIBL方法检验基因渐渗和不完全谱系分选</a><time datetime="2024-04-09T07:30:28.000Z" title="发表于 2024-04-09 15:30:28">2024-04-09</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/" title="溯祖树与基因树之间冲突的可视化"><img src="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/1.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="溯祖树与基因树之间冲突的可视化"/></a><div class="content"><a class="title" href="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/" title="溯祖树与基因树之间冲突的可视化">溯祖树与基因树之间冲突的可视化</a><time datetime="2024-01-20T11:21:18.000Z" title="发表于 2024-01-20 19:21:18">2024-01-20</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2023/10/07/phylonetwork/" title="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error"><img src="/2023/10/07/phylonetwork/1.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error"/></a><div class="content"><a class="title" href="/2023/10/07/phylonetwork/" title="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error">系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error</a><time datetime="2023-10-07T08:07:46.000Z" title="发表于 2023-10-07 16:07:46">2023-10-07</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2023/10/05/%E5%88%A9%E7%94%A8ete3%E6%89%B9%E9%87%8F%E5%AF%B9%E5%9F%BA%E5%9B%A0%E6%A0%91%E5%AE%9A%E6%A0%B9/" title="利用ete3批量对基因树定根并检测单系性">利用ete3批量对基因树定根并检测单系性</a><time datetime="2023-10-05T13:48:42.000Z" title="发表于 2023-10-05 21:48:42">2023-10-05</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2024 By 吴 田 振</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="translateLink" type="button" title="简繁转换">繁</button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="/js/tw_cn.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.umd.min.js"></script><div class="js-pjax"><script>function loadGitalk () {
  function initGitalk () {
    var gitalk = new Gitalk(Object.assign({
      clientID: 'df0416f2a2a671d6598e',
      clientSecret: '467cfdeb21916756ac8cc1a57004fd876bca7549',
      repo: 'Wu-tz.github.io',
      owner: 'Wu-tz',
      admin: ['Wu-tz'],
      id: 'e7703da65a1255f5f3927a3b52897e3f',
      updateCountCallback: commentCount
    },null))

    gitalk.render('gitalk-container')
  }

  if (typeof Gitalk === 'function') initGitalk()
  else {
    getCSS('https://cdn.jsdelivr.net/npm/gitalk/dist/gitalk.min.css')
    getScript('https://cdn.jsdelivr.net/npm/gitalk/dist/gitalk.min.js').then(initGitalk)
  }
}

function commentCount(n){
  let isCommentCount = document.querySelector('#post-meta .gitalk-comment-count')
  if (isCommentCount) {
    isCommentCount.textContent= n
  }
}

if ('Gitalk' === 'Gitalk' || !false) {
  if (false) btf.loadComment(document.getElementById('gitalk-container'), loadGitalk)
  else loadGitalk()
} else {
  function loadOtherComment () {
    loadGitalk()
  }
}</script></div><script id="click-heart" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/click-heart.min.js" async="async" mobile="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="is-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据库加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div><hr/><div class="no-result" id="local-search-results"></div><div id="local-search-stats-wrap"></div></div></div><div id="search-mask"></div><script src="/js/search/local-search.js"></script></div></div></body></html>