<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>利用cafe5进行基因家族扩张收缩分析 | TianzhenWu' Blog</title><meta name="author" content="吴 田 振"><meta name="copyright" content="吴 田 振"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="利用cafe5进行基因家族扩张收缩分析测试于2023年9月1日  基于orthofinder直系同源聚类的结果，可以看在树的特定节点上哪些基因家族发生了扩张与收缩   软件安装  123456#在单独的conda环境里安装并运行，避免环境冲突conda activate biosoftconda install -c bioconda cafe#测试成功安装cafe5 -h   准备文件包括1、带">
<meta property="og:type" content="article">
<meta property="og:title" content="利用cafe5进行基因家族扩张收缩分析">
<meta property="og:url" content="https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/index.html">
<meta property="og:site_name" content="TianzhenWu&#39; Blog">
<meta property="og:description" content="利用cafe5进行基因家族扩张收缩分析测试于2023年9月1日  基于orthofinder直系同源聚类的结果，可以看在树的特定节点上哪些基因家族发生了扩张与收缩   软件安装  123456#在单独的conda环境里安装并运行，避免环境冲突conda activate biosoftconda install -c bioconda cafe#测试成功安装cafe5 -h   准备文件包括1、带">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/1.jpg">
<meta property="article:published_time" content="2023-09-02T10:33:56.000Z">
<meta property="article:modified_time" content="2023-09-02T11:48:10.909Z">
<meta property="article:author" content="吴 田 振">
<meta property="article:tag" content="cafe5">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/1.jpg"><link rel="shortcut icon" href="/img/w-833386_640.png"><link rel="canonical" href="https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":false,"top_n_per_article":1,"unescape":false,"languages":{"hits_empty":"找不到您查询的内容：${query}","hits_stats":"共找到 ${hits} 篇文章"}},
  translate: {"defaultEncoding":2,"translateDelay":0,"msgToTraditionalChinese":"繁","msgToSimplifiedChinese":"简"},
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '天',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '利用cafe5进行基因家族扩张收缩分析',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2023-09-02 19:48:10'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
    win.getCSS = (url,id = false) => new Promise((resolve, reject) => {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = url
      if (id) link.id = id
      link.onerror = reject
      link.onload = link.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        link.onload = link.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(link)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 5.4.2"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/orca-295051_1280.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">29</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">18</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 归档</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/messageboard/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/1.jpg')"><nav id="nav"><span id="blog-info"><a href="/" title="TianzhenWu' Blog"><img class="site-icon" src="/img/w-833386_640.png"/><span class="site-name">TianzhenWu' Blog</span></a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search" href="javascript:void(0);"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 归档</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/messageboard/"><i class="fa-fw fa fa-paper-plane"></i><span> 留言板</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友链</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">利用cafe5进行基因家族扩张收缩分析</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2023-09-02T10:33:56.000Z" title="发表于 2023-09-02 18:33:56">2023-09-02</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2023-09-02T11:48:10.909Z" title="更新于 2023-09-02 19:48:10">2023-09-02</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E5%88%86%E6%9E%90%E7%AC%94%E8%AE%B0/">分析笔记</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">字数总计:</span><span class="word-count">2.3k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">阅读时长:</span><span>12分钟</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="利用cafe5进行基因家族扩张收缩分析"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h3 id="利用cafe5进行基因家族扩张收缩分析"><a href="#利用cafe5进行基因家族扩张收缩分析" class="headerlink" title="利用cafe5进行基因家族扩张收缩分析"></a>利用cafe5进行基因家族扩张收缩分析</h3><p><em>测试于2023年9月1日</em></p>
<blockquote>
<p>基于orthofinder直系同源聚类的结果，可以看在树的特定节点上哪些基因家族发生了扩张与收缩</p>
</blockquote>
<blockquote>
<p>软件安装</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#在单独的conda环境里安装并运行，避免环境冲突</span></span><br><span class="line">conda activate biosoft</span><br><span class="line">conda install -c bioconda cafe</span><br><span class="line"></span><br><span class="line"><span class="comment">#测试成功安装</span></span><br><span class="line">cafe5 -h</span><br></pre></td></tr></table></figure>

<blockquote>
<p>准备文件包括1、带有分歧时间的树，由MCMCtree产生；2、直系同源基因家族的聚类情况，由orthofinder结果产生；</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#step 1 准备树文件，FigTree.tre来自MCMCtree结果</span></span><br><span class="line"></span><br><span class="line">grep  <span class="string">&quot;UTREE 1 =&quot;</span> FigTree.tre | sed -E -e <span class="string">&quot;s/\[[^]]*\]//g&quot;</span> -e <span class="string">&quot;s/[ \t]//g&quot;</span> -e <span class="string">&quot;/^$/d&quot;</span> -e <span class="string">&quot;s/UTREE1=//&quot;</span> &gt; tree.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">#step 2 用orthofinder2的结果文件Orthogroups.GeneCount.tsv转换成gene_families.txt文件，文件路径在/home/tianzhen/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups</span></span><br><span class="line"></span><br><span class="line">awk -v OFS=<span class="string">&quot;\t&quot;</span> <span class="string">&#x27;&#123;$NF=null;print $1,$0&#125;&#x27;</span> Orthogroups.GeneCount.tsv |sed -E -e <span class="string">&#x27;s/Orthogroup/desc/&#x27;</span> -e <span class="string">&#x27;s/_[^\t]+//g&#x27;</span> &gt; orthomcl2cafe.tab</span><br><span class="line"> </span><br><span class="line"><span class="comment">#或者使用perl脚本提取orthomcl2cafe.tab文件，Orthogroups.txt 文件所在路径为/OrthoFinder/Results_Nov22/Orthogroups/Orthogroups.txt</span></span><br><span class="line"></span><br><span class="line">perl orthoMCL2cafe.pl Orthogroups.txt &gt; orthomcl2cafe.tab</span><br><span class="line"></span><br><span class="line"><span class="comment">#step 3 剔除不同物种间拷贝数差异过大的基因家族，否则会报错</span></span><br><span class="line"></span><br><span class="line">python ~/scripts/cafetutorial_clade_and_size_filter.py -i orthomcl2cafe.tab -o gene_family_filter.txt -s</span><br></pre></td></tr></table></figure>

<blockquote>
<p>运行cafe5软件</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#step 1 首先评估lambda，此步应该可以省略，在运行中不添加-l参数，则会自动评估lambda值</span></span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -o singlelambda</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可能因为某些基因家族数量过大会报错，报错内容为：</span></span><br><span class="line">Families with largest size differentials:</span><br><span class="line">OG0000385: 97</span><br><span class="line">OG0000296: 97</span><br><span class="line">OG0000029: 96</span><br><span class="line">OG0000030: 95</span><br><span class="line">OG0000016: 93</span><br><span class="line">OG0000142: 92</span><br><span class="line">OG0000232: 91</span><br><span class="line">OG0000035: 91</span><br><span class="line">OG0000155: 89</span><br><span class="line">OG0000115: 89</span><br><span class="line">OG0000027: 88</span><br><span class="line">OG0000079: 86</span><br><span class="line">OG0000034: 83</span><br><span class="line">OG0000044: 82</span><br><span class="line">OG0000055: 80</span><br><span class="line">OG0000067: 79</span><br><span class="line">OG0000943: 78</span><br><span class="line">OG0000050: 77</span><br><span class="line">OG0000110: 76</span><br><span class="line">OG0000088: 76</span><br><span class="line">You may want to try removing the top few families with the largest difference</span><br><span class="line">between the max and min counts and <span class="keyword">then</span> re-run the analysis.</span><br><span class="line"></span><br><span class="line">Failed to initialize any reasonable values</span><br><span class="line"></span><br><span class="line"><span class="comment"># 将上述基因家族写入文件，准备从orthomcl2cafe.tab中剔除掉</span></span><br><span class="line">vi families_largest.txt</span><br><span class="line">sed -i <span class="string">&#x27;s/:.*//g&#x27;</span> families_largest.txt</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> `cat families_largest.txt`;<span class="keyword">do</span> sed -i <span class="string">&quot;/<span class="variable">$i</span>/d&quot;</span> orthomcl2cafe.tab;<span class="keyword">done</span></span><br><span class="line">(grep -v -f family.remove.id.txt input.tab &gt;input.tab.new)</span><br><span class="line"><span class="comment"># 重新运行，评估lambda值</span></span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -o singlelambda</span><br><span class="line"></span><br><span class="line"><span class="comment">#step 2 调整k参数2-5,多次运行，选择最优k值的结果</span></span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -k 2 -l 0.0001 -o k2p</span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -k 3 -l 0.0001 -o k3p</span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -k 4 -l 0.0001 -o k4p</span><br><span class="line">cafe5 -i orthomcl2cafe.tab -t tree.txt -p -k 5 -l 0.0001 -o k5p</span><br></pre></td></tr></table></figure>

<blockquote>
<p>在结果中的Gamma_results.txt文件里查看lnL值，选择该值最大的k值运行结果作为后续分析的文件</p>
<p>接下来提取感兴趣节点的基因家族序列</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#在Gamma_change.tab文件里提取节点35的扩张的家族，位于第37列</span></span><br><span class="line">cat ../Gamma_change.tab |cut -f1,37|grep <span class="string">&quot;+[1-9]&quot;</span> &gt; cr.expanded    </span><br><span class="line"></span><br><span class="line"><span class="comment">#根据sample ID和编号提取sample分支的基因家族显著扩张或收缩的基因家族树（Gamma_asr.tre文件中默认以p&lt;0.05为标准判断变化是否显著）</span></span><br><span class="line">grep <span class="string">&quot;&lt;35&gt;\*&quot;</span> ../Gamma_asr.tre &gt; cr_significant_trees.tre    </span><br><span class="line"></span><br><span class="line"><span class="comment">#提取sample分支显著变化的OG IDs （默认以p&lt;0.05为标准）</span></span><br><span class="line">grep -E -o <span class="string">&quot;OG[0-9]+&quot;</span> cr_significant_trees.tre &gt; cr_significant.ogs  </span><br><span class="line"></span><br><span class="line"><span class="comment"># 以p&lt;0.05为标准提取所有显著扩张或收缩的orthogroupsID（根据情况调整，常用p&lt;0.05或p&lt;0.01）</span></span><br><span class="line">awk <span class="string">&#x27;$2 &lt;0.05 &#123;print $1&#125;&#x27;</span> ../Gamma_family_results.txt &gt;p0.05_significant.ogs </span><br><span class="line"></span><br><span class="line"><span class="comment"># 提取以p&lt;0.05为标准判断显著性的sample分支基因家族显著变化的OG IDs</span></span><br><span class="line">grep -f cr_significant.ogs p0.05_significant.ogs &gt; cr_p0.05_significant.ogs </span><br><span class="line"></span><br><span class="line"><span class="comment">#提取显著扩张的sample物种的orthogroupsID</span></span><br><span class="line">grep -f cr_p0.05_significant.ogs cr.expanded |cut -f1 &gt; cr.expanded.significant</span><br><span class="line"></span><br><span class="line"><span class="comment">#提取显著扩张的基因列表，假设基因ID的前缀是amil</span></span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep <span class="string">&quot;amil&quot;</span> |sort -k 1.3n |uniq &gt; cr.amil.expanded.significant.genes</span><br><span class="line"><span class="comment">#祖先节点则将所有物种的序列都提出来</span></span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep adig |sort -k 1.3n |uniq &gt; cr.adig.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep aequ |sort -k 1.3n |uniq &gt; cr.aequ.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep afen |sort -k 1.3n |uniq &gt; cr.afen.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep alor |sort -k 1.3n |uniq &gt; cr.alor.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep amil |sort -k 1.3n |uniq &gt; cr.amil.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep apall |sort -k 1.3n |uniq &gt; cr.apall.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep atene |sort -k 1.3n |uniq &gt; cr.atene.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep atenu |sort -k 1.3n |uniq &gt; cr.atenu.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep dgig |sort -k 1.3n |uniq &gt; cr.dgig.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep disc |sort -k 1.3n |uniq &gt; cr.disc.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep edia |sort -k 1.3n |uniq &gt; cr.edia.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep fung |sort -k 1.3n |uniq &gt; cr.fung.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep gasp |sort -k 1.3n |uniq &gt; cr.gasp.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep gfas |sort -k 1.3n |uniq &gt; cr.gfas.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep hvul |sort -k 1.3n |uniq &gt; cr.hvul.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep nvec |sort -k 1.3n |uniq &gt; cr.nvec.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep ofav |sort -k 1.3n |uniq &gt; cr.ofav.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep pdae |sort -k 1.3n |uniq &gt; cr.pdae.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep pdam |sort -k 1.3n |uniq &gt; cr.pdam.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep plut |sort -k 1.3n |uniq &gt; cr.plut.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep pspe |sort -k 1.3n |uniq &gt; cr.pspe.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep pverr |sort -k 1.3n |uniq &gt; cr.pverr.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep rmue |sort -k 1.3n |uniq &gt; cr.rmue.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep spis |sort -k 1.3n |uniq &gt; cr.spis.expanded.significant.genes</span><br><span class="line">grep -f cr.expanded.significant ~/2023-redo-cafe/OrthoFinder/Results_Aug27/Orthogroups/Orthogroups.txt|sed <span class="string">&quot;s/ /\n/g&quot;</span>|grep xeni |sort -k 1.3n |uniq &gt; cr.xeni.expanded.significant.genes</span><br><span class="line"></span><br><span class="line"><span class="comment">#提取所有显著扩张的基因序列，用于基因注释</span></span><br><span class="line">seqkit grep -f ../cr.adig.expanded.significant.genes /home/tianzhen/2023-redo-cafe/adig.fasta &gt; cr.adig.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.aequ.expanded.significant.genes /home/tianzhen/2023-redo-cafe/aequ.fasta &gt; cr.aequ.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.afen.expanded.significant.genes /home/tianzhen/2023-redo-cafe/afen.fasta &gt; cr.afen.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.alor.expanded.significant.genes /home/tianzhen/2023-redo-cafe/alor.fasta &gt; cr.alor.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.amil.expanded.significant.genes /home/tianzhen/2023-redo-cafe/amil.fasta &gt; cr.amil.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.apall.expanded.significant.genes /home/tianzhen/2023-redo-cafe/apall.fasta &gt; cr.apall.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.atene.expanded.significant.genes /home/tianzhen/2023-redo-cafe/atene.fasta &gt; cr.atene.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.atenu.expanded.significant.genes /home/tianzhen/2023-redo-cafe/atenu.fasta &gt; cr.atenu.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.dgig.expanded.significant.genes /home/tianzhen/2023-redo-cafe/dgig.fasta &gt; cr.dgig.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.disc.expanded.significant.genes /home/tianzhen/2023-redo-cafe/disc.fasta &gt; cr.disc.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.edia.expanded.significant.genes /home/tianzhen/2023-redo-cafe/edia.fasta &gt; cr.edia.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.fung.expanded.significant.genes /home/tianzhen/2023-redo-cafe/fung.fasta &gt; cr.fung.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.gasp.expanded.significant.genes /home/tianzhen/2023-redo-cafe/gasp.fasta &gt; cr.gasp.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.gfas.expanded.significant.genes /home/tianzhen/2023-redo-cafe/gfas.fasta &gt; cr.gfas.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.hvul.expanded.significant.genes /home/tianzhen/2023-redo-cafe/hvul.fasta &gt; cr.hvul.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.nvec.expanded.significant.genes /home/tianzhen/2023-redo-cafe/nvec.fasta &gt; cr.nvec.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.ofav.expanded.significant.genes /home/tianzhen/2023-redo-cafe/ofav.fasta &gt; cr.ofav.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.pdae.expanded.significant.genes /home/tianzhen/2023-redo-cafe/pdae.fasta &gt; cr.pdae.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.pdam.expanded.significant.genes /home/tianzhen/2023-redo-cafe/pdam.fasta &gt; cr.pdam.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.plut.expanded.significant.genes /home/tianzhen/2023-redo-cafe/plut.fasta &gt; cr.plut.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.pspe.expanded.significant.genes /home/tianzhen/2023-redo-cafe/pspe.fasta &gt; cr.pspe.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.pverr.expanded.significant.genes /home/tianzhen/2023-redo-cafe/pverr.fasta &gt; cr.pverr.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.rmue.expanded.significant.genes /home/tianzhen/2023-redo-cafe/rmue.fasta &gt; cr.rmue.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.spis.expanded.significant.genes /home/tianzhen/2023-redo-cafe/spis.fasta &gt; cr.spis.expanded.significant.pep.fas</span><br><span class="line">seqkit grep -f ../cr.xeni.expanded.significant.genes /home/tianzhen/2023-redo-cafe/xeni.fasta &gt; cr.xeni.expanded.significant.pep.fas</span><br><span class="line"></span><br><span class="line"><span class="comment">#将上面的序列合为一个fasta文件，用于基因功能注释</span></span><br><span class="line">cat *.fas &gt; all_sequences_for_anno.fas</span><br></pre></td></tr></table></figure>

<p>参考</p>
<p><a target="_blank" rel="noopener" href="https://github.com/hahnlab/CAFE5">hahnlab&#x2F;CAFE5: Version 5 of the CAFE phylogenetics software (github.com)</a></p>
<p><a target="_blank" rel="noopener" href="https://yanzhongsino.github.io/2021/10/29/bioinfo_gene.family_CAFE5/">分析基因家族扩张和收缩 —— CAFE5 | 生信技工 (yanzhongsino.github.io)</a></p>
<link rel="stylesheet" href="/css/spoiler.css" type="text/css"><script src="/js/spoiler.js" type="text/javascript" async></script></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="https://wu-tz.github.io">吴 田 振</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/">https://wu-tz.github.io/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="https://wu-tz.github.io" target="_blank">TianzhenWu' Blog</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/cafe5/">cafe5</a></div><div class="post_share"><div class="social-share" data-image="/2023/09/02/%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90/1.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i> 打赏</div><div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="/img/wechat.jpg" target="_blank"><img class="post-qr-code-img" src="/img/wechat.jpg" alt="wechat"/></a><div class="post-qr-code-desc">wechat</div></li><li class="reward-item"><a href="/img/alipay.jpg" target="_blank"><img class="post-qr-code-img" src="/img/alipay.jpg" alt="alipay"/></a><div class="post-qr-code-desc">alipay</div></li></ul></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2023/10/05/%E5%88%A9%E7%94%A8ete3%E6%89%B9%E9%87%8F%E5%AF%B9%E5%9F%BA%E5%9B%A0%E6%A0%91%E5%AE%9A%E6%A0%B9/" title="利用ete3批量对基因树定根并检测单系性"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">利用ete3批量对基因树定根并检测单系性</div></div></a></div><div class="next-post pull-right"><a href="/2023/04/04/ete3/" title="ete3"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">ete3</div></div></a></div></nav><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="gitalk-container"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/orca-295051_1280.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">吴 田 振</div><div class="author-info__description">努力发一区</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">29</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">18</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><a id="card-info-btn" href="https://wu-tz.github.io/"><i class="iconfont icon-CSDN"></i><span>收藏我啊</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/Wu-tz" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="mailto:wtzsdut@163.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a><a class="social-icon" href="tencent://AddContact/?fromId=45&amp;fromSubId=1&amp;subcmd=all&amp;uin=2312164805&amp;website=www.oicqzone.com" target="_blank" title="QQ"><i class="fab fa-qq"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">在这里记录的是一些随笔，包括文件处理技巧、软件使用、生物知识等，作为我的工作以及生活记录。欢迎您的访问，祝工作顺利！喜欢的话欢迎收藏~</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%88%A9%E7%94%A8cafe5%E8%BF%9B%E8%A1%8C%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%89%A9%E5%BC%A0%E6%94%B6%E7%BC%A9%E5%88%86%E6%9E%90"><span class="toc-number">1.</span> <span class="toc-text">利用cafe5进行基因家族扩张收缩分析</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/07/17/genome-anno/" title="基因组注释——适合初学者的流程分解版">基因组注释——适合初学者的流程分解版</a><time datetime="2024-07-17T13:06:54.000Z" title="发表于 2024-07-17 21:06:54">2024-07-17</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/" title="QuIBL方法检验基因渐渗和不完全谱系分选"><img src="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/2024-04-09%20154337.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="QuIBL方法检验基因渐渗和不完全谱系分选"/></a><div class="content"><a class="title" href="/2024/04/09/QuIBL%E6%96%B9%E6%B3%95%E6%A3%80%E9%AA%8C%E5%9F%BA%E5%9B%A0%E6%B8%90%E6%B8%97%E5%92%8C%E4%B8%8D%E5%AE%8C%E5%85%A8%E8%B0%B1%E7%B3%BB%E5%88%86%E9%80%89/" title="QuIBL方法检验基因渐渗和不完全谱系分选">QuIBL方法检验基因渐渗和不完全谱系分选</a><time datetime="2024-04-09T07:30:28.000Z" title="发表于 2024-04-09 15:30:28">2024-04-09</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/" title="溯祖树与基因树之间冲突的可视化"><img src="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/1.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="溯祖树与基因树之间冲突的可视化"/></a><div class="content"><a class="title" href="/2024/01/20/%E6%BA%AF%E7%A5%96%E6%A0%91%E4%B8%8E%E5%9F%BA%E5%9B%A0%E6%A0%91%E4%B9%8B%E9%97%B4%E5%86%B2%E7%AA%81%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96/" title="溯祖树与基因树之间冲突的可视化">溯祖树与基因树之间冲突的可视化</a><time datetime="2024-01-20T11:21:18.000Z" title="发表于 2024-01-20 19:21:18">2024-01-20</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2023/10/07/phylonetwork/" title="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error"><img src="/2023/10/07/phylonetwork/1.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error"/></a><div class="content"><a class="title" href="/2023/10/07/phylonetwork/" title="系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error">系统发育网络推断软件PhyloNetworks和Phylonet的使用及遇到的error</a><time datetime="2023-10-07T08:07:46.000Z" title="发表于 2023-10-07 16:07:46">2023-10-07</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2023/10/05/%E5%88%A9%E7%94%A8ete3%E6%89%B9%E9%87%8F%E5%AF%B9%E5%9F%BA%E5%9B%A0%E6%A0%91%E5%AE%9A%E6%A0%B9/" title="利用ete3批量对基因树定根并检测单系性">利用ete3批量对基因树定根并检测单系性</a><time datetime="2023-10-05T13:48:42.000Z" title="发表于 2023-10-05 21:48:42">2023-10-05</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2024 By 吴 田 振</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="translateLink" type="button" title="简繁转换">繁</button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="/js/tw_cn.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.umd.min.js"></script><div class="js-pjax"><script>function loadGitalk () {
  function initGitalk () {
    var gitalk = new Gitalk(Object.assign({
      clientID: 'df0416f2a2a671d6598e',
      clientSecret: '467cfdeb21916756ac8cc1a57004fd876bca7549',
      repo: 'Wu-tz.github.io',
      owner: 'Wu-tz',
      admin: ['Wu-tz'],
      id: '67bd58c819476a96540b69c0d5996339',
      updateCountCallback: commentCount
    },null))

    gitalk.render('gitalk-container')
  }

  if (typeof Gitalk === 'function') initGitalk()
  else {
    getCSS('https://cdn.jsdelivr.net/npm/gitalk/dist/gitalk.min.css')
    getScript('https://cdn.jsdelivr.net/npm/gitalk/dist/gitalk.min.js').then(initGitalk)
  }
}

function commentCount(n){
  let isCommentCount = document.querySelector('#post-meta .gitalk-comment-count')
  if (isCommentCount) {
    isCommentCount.textContent= n
  }
}

if ('Gitalk' === 'Gitalk' || !false) {
  if (false) btf.loadComment(document.getElementById('gitalk-container'), loadGitalk)
  else loadGitalk()
} else {
  function loadOtherComment () {
    loadGitalk()
  }
}</script></div><script id="click-heart" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/click-heart.min.js" async="async" mobile="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="is-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据库加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div><hr/><div class="no-result" id="local-search-results"></div><div id="local-search-stats-wrap"></div></div></div><div id="search-mask"></div><script src="/js/search/local-search.js"></script></div></div></body></html>